---
layout: default
feature_image: images/microbes.jpg
---

<h1>Quantifying the varying harvest of fermentation products from the human gut microbiota</h1>
Markus Arnoldini, Richa Sharma<sup>*</sup>, Claudia Moresi<sup>*</sup>, Griffin Chure, Julien Chabbey, Emma Slack, and
Jonas Cremer

<h2>Abstract</h2>
Fermentation products released by the gut microbiota provide energy and regulatory functions to the host. Yet, little is known about the magnitude of this metabolic flux and its quantitative dependence on diet and microbiome composition. Here, we establish orthogonal approaches to consistently quantify this flux, integrating data on bacterial metabolism, digestive physiology, and metagenomics. From the nutrients fueling microbiota growth, most carbon ends up in fermentation products and is absorbed by the host. This harvest varies strongly with the amount of complex dietary carbohydrates and is largely independent of bacterial mucin and protein utilization. It covers 2-5% of human energy demand for western, and up to 10% for non-western diets. Microbiota composition has little impact on the total harvest but determines the amount of specific fermentation products. This consistent quantification of metabolic fluxes by our analysis framework is crucial to elucidate the gut microbiota’s mechanistic functions in health and disease.

<h2>Interactive Fermentation Product Secretion Explorer</h2>

This figure permits an exploration of the distribution of our estimated fermentation product 
secretion from published studies of human microbiome composition. The top panel allows 
you to select between different published studies and filter on individuals within 
different age categories and health status. The grey histogram (right) illustrates
what fraction of the total microbial biomass is represented by the bacterial species 
we experimentally examined in this study.


The middle panel allows you to either 
i) directly adjust how much microbiota available carbohydrate reaches the lower intestine,
ii) directly adjust how much starch and fibers are consumed, or iii) select different 
reference diets as described in the main text of the paper. 


The bottom panel shows the distributions of expected microbial biomass and 
fermentation products. Additionally, we show the total daily energy content of all 
secreted fermentation products. 
<hr/>

<center>

{% include interactive_excretion.html %}


</center>


### Referenced Data Sets 
1. Asnicar, F et al., <a href="https://doi.org/10.1128/mSystems.00164-16">"Studying Vertical Microbiome Transmission from Mothers to Infants by Strain-Level Metagenomic Profiling"</a>. mSystems. 2017 Jan 17;2(1):e00164-16. doi: 10.1128/mSystems.00164-16. eCollection 2017 Jan-Feb.
2. Asnicar, F et al., <a href="https://doi.org/10.1038/s41591-020-01183-8">"Microbiome connections with host metabolism and habitual diet from 1,098 deeply phenotyped individuals"</a>. Nat Med. 2021 Feb;27(2):321-332. doi: 10.1038/s41591-020-01183-8. Epub 2021 Jan 11.
3. BŠckhed, F et al., <a href="https://doi.org/10.1016/j.chom.2015.04.004">"Dynamics and Stabilization of the Human Gut Microbiome during the First Year of Life"</a>. Cell Host Microbe. 2015 May 13;17(5):690-703. doi: 10.1016/j.chom.2015.04.004.
4. Bengtsson-Palme, J et al., <a href="https://doi.org/10.1128/AAC.00933-15">"The Human Gut Microbiome as a Transporter of Antibiotic Resistance Genes between Continents"</a>. Antimicrob Agents Chemother. 2015 Oct;59(10):6551-60. doi: 10.1128/AAC.00933-15. Epub 2015 Aug 10.
5. Brito, IL et al., <a href="https://doi.org/10.1038/nature18927">"Mobile genes in the human microbiome are structured from global to individual scales"</a>. Nature. 2016 Jul 21;535(7612):435-439. doi: 10.1038/nature18927. Epub 2016 Jul 13.
6. Brooks, B et al., <a href="https://doi.org/10.1038/s41467-017-02018-w">"Strain-resolved analysis of hospital rooms and infants reveals overlap between the human and room microbiome"</a>. Nat Commun. 2017 Nov 27;8(1):1814. doi: 10.1038/s41467-017-02018-w.
7. Wen, C et al., <a href="https://doi.org/10.1186/s13059-017-1271-6">"Quantitative metagenomics reveals unique gut microbiome biomarkers in ankylosing spondylitis"</a>. Genome Biol. 2017 Jul 27;18(1):142. doi: 10.1186/s13059-017-1271-6.
8. Chu, DM et al., <a href="https://doi.org/10.1038/nm.4272">"Maturation of the infant microbiome community structure and function across multiple body sites and in relation to mode of delivery"</a>. Nat Med. 2017 Mar;23(3):314-326. doi: 10.1038/nm.4272. Epub 2017 Jan 23.
9. Costea, PI et al., <a href="https://doi.org/10.15252/msb.20177589">"Subspecies in the global human gut microbiome"</a>. Mol Syst Biol. 2017 Dec 14;13(12):960. doi: 10.15252/msb.20177589.
10. David, LA et al., <a href="https://doi.org/10.1128/mBio.00381-15">"Gut microbial succession follows acute secretory diarrhea in humans"</a>. mBio. 2015 May 19;6(3):e00381-15. doi: 10.1128/mBio.00381-15.
11. De, Filippis, F et al., <a href="https://doi.org/10.1016/j.chom.2019.01.004">"Distinct Genetic and Functional Traits of Human Intestinal Prevotella copri Strains Are Associated with Different Habitual Diets"</a>. Cell Host Microbe. 2019 Mar 13;25(3):444-453.e3. doi: 10.1016/j.chom.2019.01.004. Epub 2019 Feb 21.
12. Dhakan, DB et al., <a href="https://doi.org/10.1093/gigascience/giz004">"The unique composition of Indian gut microbiome, gene catalogue, and associated fecal metabolome deciphered using multi-omics approaches"</a>. Gigascience. 2019 Mar 1;8(3):giz004. doi: 10.1093/gigascience/giz004.
13. Feng, Q et al., <a href="https://doi.org/10.1038/ncomms7528">"Gut microbiome development along the colorectal adenoma-carcinoma sequence"</a>. Nat Commun. 2015 Mar 11;6:6528. doi: 10.1038/ncomms7528.
14. Ferretti, P et al., <a href="https://doi.org/10.1016/j.chom.2018.06.005">"Mother-to-Infant Microbial Transmission from Different Body Sites Shapes the Developing Infant Gut Microbiome"</a>. Cell Host Microbe. 2018 Jul 11;24(1):133-145.e5. doi: 10.1016/j.chom.2018.06.005.
15. Gopalakrishnan, V et al., <a href="https://doi.org/10.1126/science.aan4236">"Gut microbiome modulates response to anti-PD-1 immunotherapy in melanoma patients"</a>. Science. 2018 Jan 5;359(6371):97-103. doi: 10.1126/science.aan4236. Epub 2017 Nov 2.
16. Dhakan, DB et al., <a href="https://doi.org/10.1093/gigascience/giz004">"The unique composition of Indian gut microbiome, gene catalogue, and associated fecal metabolome deciphered using multi-omics approaches"</a>. Gigascience. 2019 Mar 1;8(3):giz004. doi: 10.1093/gigascience/giz004.
17. Hall, AB et al., <a href="https://doi.org/10.1186/s13073-017-0490-5">"A novel Ruminococcus gnavus clade enriched in inflammatory bowel disease patients"</a>. Genome Med. 2017 Nov 28;9(1):103. doi: 10.1186/s13073-017-0490-5.
18. Hannigan, GD et al., <a href="https://doi.org/10.1128/mBio.02248-18">"Diagnostic Potential and Interactive Dynamics of the Colorectal Cancer Virome"</a>. mBio. 2018 Nov 20;9(6):e02248-18. doi: 10.1128/mBio.02248-18.
19. Hansen, LBS et al., <a href="https://doi.org/10.1038/s41467-018-07019-x">"A low-gluten diet induces changes in the intestinal microbiome of healthy Danish adults"</a>. Nat Commun. 2018 Nov 13;9(1):4630. doi: 10.1038/s41467-018-07019-x.
20. Heintz-Buschart, A et al., <a href="https://doi.org/10.1038/nmicrobiol.2016.180">"Integrated multi-omics of the human gut microbiome in a case study of familial type 1 diabetes"</a>. Nat Microbiol. 2016 Oct 10;2:16180. doi: 10.1038/nmicrobiol.2016.180.
21. Human Microbiome Project Consortium., <a href="https://doi.org/10.1038/nature11234">"Structure, function and diversity of the healthy human microbiome"</a>. Nature. 2012 Jun 13;486(7402):207-14. doi: 10.1038/nature11234.
22. Schirmer, M et al., <a href="https://doi.org/10.1038/s41564-017-0089-z">"Dynamics of metatranscription in the inflammatory bowel disease gut microbiome"</a>. Nat Microbiol. 2018 Mar;3(3):337-346. doi: 10.1038/s41564-017-0089-z. Epub 2018 Jan 8.
23. Zhou, W et al., <a href="https://doi.org/10.1038/s41586-019-1236-x">"Longitudinal multi-omics of host-microbe dynamics in prediabetes"</a>. Nature. 2019 May;569(7758):663-671. doi: 10.1038/s41586-019-1236-x. Epub 2019 May 29.
24. Ijaz, UZ et al., <a href="https://doi.org/10.1371/journal.pone.0172605">"The distinct features of microbial 'dysbiosis' of Crohn's disease do not occur to the same extent in their unaffected, genetically-linked kindred"</a>. PLoS One. 2017 Feb 21;12(2):e0172605. doi: 10.1371/journal.pone.0172605. eCollection 2017.
25. Jie, Z et al., <a href="https://doi.org/10.1038/s41467-017-00900-1">"The gut microbiome in atherosclerotic cardiovascular disease"</a>. Nat Commun. 2017 Oct 10;8(1):845. doi: 10.1038/s41467-017-00900-1.
26. Karlsson, FH et al., <a href="https://doi.org/10.1038/nature12198">"Gut metagenome in European women with normal, impaired and diabetic glucose control"</a>. Nature. 2013 Jun 6;498(7452):99-103. doi: 10.1038/nature12198. Epub 2013 May 29.
27. Kaur, K et al., <a href="https://doi.org/10.1371/journal.pone.0231197">"Metagenomics analysis reveals features unique to Indian distal gut microbiota"</a>. PLoS One. 2020 Apr 8;15(4):e0231197. doi: 10.1371/journal.pone.0231197. eCollection 2020.
28. Keohane, DM et al., <a href="https://doi.org/10.1038/s41591-020-0963-8">"Microbiome and health implications for ethnic minorities after enforced lifestyle changes"</a>. Nat Med. 2020 Jul;26(7):1089-1095. doi: 10.1038/s41591-020-0963-8. Epub 2020 Jul 6.
29. Kieser, S et al., <a href="https://doi.org/10.1111/1462-2920.14274">"Bangladeshi children with acute diarrhoea show faecal microbiomes with increased Streptococcus abundance, irrespective of diarrhoea aetiology"</a>. Environ Microbiol. 2018 Jun;20(6):2256-2269. doi: 10.1111/1462-2920.14274. Epub 2018 Aug 7.
30. Kostic, AD et al., <a href="https://doi.org/10.1016/j.chom.2015.01.001">"The dynamics of the human infant gut microbiome in development and in progression toward type 1 diabetes"</a>. Cell Host Microbe. 2015 Feb 11;17(2):260-73. doi: 10.1016/j.chom.2015.01.001. Epub 2015 Feb 5.
31. Le, Chatelier, E et al., <a href="https://doi.org/10.1038/nature12506">"Richness of human gut microbiome correlates with metabolic markers"</a>. Nature. 2013 Aug 29;500(7464):541-6. doi: 10.1038/nature12506.
32. Zhernakova, A et al., <a href="https://doi.org/10.1126/science.aad3369">"Population-based metagenomics analysis reveals markers for gut microbiome composition and diversity"</a>. Science. 2016 Apr 29;352(6285):565-9. doi: 10.1126/science.aad3369. Epub 2016 Apr 28.
33. Li, J et al., <a href="https://doi.org/10.1038/nbt.2942">"An integrated catalog of reference genes in the human gut microbiome"</a>. Nat Biotechnol. 2014 Aug;32(8):834-41. doi: 10.1038/nbt.2942. Epub 2014 Jul 6.
34. Li, J et al., <a href="https://doi.org/10.1186/s40168-016-0222-x">"Gut microbiota dysbiosis contributes to the development of hypertension"</a>. Microbiome. 2017 Feb 1;5(1):14. doi: 10.1186/s40168-016-0222-x.
35. Li, SS et al., <a href="https://doi.org/10.1126/science.aad8852">"Durable coexistence of donor and recipient strains after fecal microbiota transplantation"</a>. Science. 2016 Apr 29;352(6285):586-9. doi: 10.1126/science.aad8852.
36. Liu, W et al., <a href="https://doi.org/10.1038/srep34826">"Unique Features of Ethnic Mongolian Gut Microbiome revealed by metagenomic analysis"</a>. Sci Rep. 2016 Oct 6;6:34826. doi: 10.1038/srep34826.
37. Lokmer, A et al., <a href="https://doi.org/10.1371/journal.pone.0211139">"Use of shotgun metagenomics for the identification of protozoa in the gut microbiota of healthy individuals from worldwide populations with various industrialization levels"</a>. PLoS One. 2019 Feb 6;14(2):e0211139. doi: 10.1371/journal.pone.0211139. eCollection 2019.
38. Loman, NJ et al., <a href="https://doi.org/10.1001/jama.2013.3231">"A culture-independent sequence-based metagenomics approach to the investigation of an outbreak of Shiga-toxigenic Escherichia coli O104:H4"</a>. JAMA. 2013 Apr 10;309(14):1502-10. doi: 10.1001/jama.2013.3231.
39. Loomba, R et al., <a href="https://doi.org/10.1016/j.cmet.2017.04.001">"Gut Microbiome-Based Metagenomic Signature for Non-invasive Detection of Advanced Fibrosis in Human Nonalcoholic Fatty Liver Disease"</a>. Cell Metab. 2017 May 2;25(5):1054-1062.e5. doi: 10.1016/j.cmet.2017.04.001.
40. Louis, S et al., <a href="https://doi.org/10.1371/journal.pone.0149564">"Characterization of the Gut Microbial Community of Obese Patients Following a Weight-Loss Intervention Using Whole Metagenome Shotgun Sequencing"</a>. PLoS One. 2016 Feb 26;11(2):e0149564. doi: 10.1371/journal.pone.0149564. eCollection 2016.
41. Matson, V et al., <a href="https://doi.org/10.1126/science.aao3290">"The commensal microbiome is associated with anti-PD-1 efficacy in metastatic melanoma patients"</a>. Science. 2018 Jan 5;359(6371):104-108. doi: 10.1126/science.aao3290.
42. Mehta, RS et al., <a href="https://doi.org/10.1038/s41564-017-0096-0">"Stability of the human faecal microbiome in a cohort of adult men"</a>. Nat Microbiol. 2018 Mar;3(3):347-355. doi: 10.1038/s41564-017-0096-0. Epub 2018 Jan 15.
43. Nagy-Szakal, D et al., <a href="https://doi.org/10.1186/s40168-017-0261-y">"Fecal metagenomic profiles in subgroups of patients with myalgic encephalomyelitis/chronic fatigue syndrome"</a>. Microbiome. 2017 Apr 26;5(1):44. doi: 10.1186/s40168-017-0261-y.
44. Nielsen, HB et al., <a href="https://doi.org/10.1038/nbt.2939">"Identification and assembly of genomes and genetic elements in complex metagenomic samples without using reference genomes"</a>. Nat Biotechnol. 2014 Aug;32(8):822-8. doi: 10.1038/nbt.2939. Epub 2014 Jul 6.
45. Obregon-Tito, AJ et al., <a href="https://doi.org/10.1038/ncomms7505">"Subsistence strategies in traditional societies distinguish gut microbiomes"</a>. Nat Commun. 2015 Mar 25;6:6505. doi: 10.1038/ncomms7505.
46. Olm, MR et al., <a href="https://doi.org/10.1101/gr.213256.116">"Identical bacterial populations colonize premature infant gut, skin, and oral microbiomes and exhibit different in situ growth rates"</a>. Genome Res. 2017 Apr;27(4):601-612. doi: 10.1101/gr.213256.116. Epub 2017 Jan 10.
47. Pasolli, E et al., <a href="https://doi.org/10.1016/j.cell.2019.01.001">"Extensive Unexplored Human Microbiome Diversity Revealed by Over 150,000 Genomes from Metagenomes Spanning Age, Geography, and Lifestyle"</a>. Cell. 2019 Jan 24;176(3):649-662.e20. doi: 10.1016/j.cell.2019.01.001. Epub 2019 Jan 17.
48. Pehrsson, EC et al., <a href="https://doi.org/10.1038/nature17672">"Interconnected microbiomes and resistomes in low-income human habitats"</a>. Nature. 2016 May 12;533(7602):212-6. doi: 10.1038/nature17672.
49. Qin, J et al., <a href="https://doi.org/10.1038/nature11450">"A metagenome-wide association study of gut microbiota in type 2 diabetes"</a>. Nature. 2012 Oct 4;490(7418):55-60. doi: 10.1038/nature11450. Epub 2012 Sep 26.
50. Qin, N et al., <a href="https://doi.org/10.1038/nature13568">"Alterations of the human gut microbiome in liver cirrhosis"</a>. Nature. 2014 Sep 4;513(7516):59-64. doi: 10.1038/nature13568. Epub 2014 Jul 23.
51. Rampelli, S et al., <a href="https://doi.org/10.1016/j.cub.2015.04.055">"Metagenome Sequencing of the Hadza Hunter-Gatherer Gut Microbiota"</a>. Curr Biol. 2015 Jun 29;25(13):1682-93. doi: 10.1016/j.cub.2015.04.055. Epub 2015 May 14.
52. Raymond, F et al., <a href="https://doi.org/10.1038/ismej.2015.148">"The initial state of the human gut microbiome determines its reshaping by antibiotics"</a>. ISME J. 2016 Mar;10(3):707-20. doi: 10.1038/ismej.2015.148. Epub 2015 Sep 11.
53. Rosa, BA et al., <a href="https://doi.org/10.1186/s40168-018-0416-5">"Differential human gut microbiome assemblages during soil-transmitted helminth infections in Indonesia and Liberia"</a>. Microbiome. 2018 Feb 28;6(1):33. doi: 10.1186/s40168-018-0416-5.
54. Rubel, MA et al., <a href="https://doi.org/10.1186/s13059-020-02020-4">"Lifestyle and the presence of helminths is associated with gut microbiome composition in Cameroonians"</a>. Genome Biol. 2020 May 25;21(1):122. doi: 10.1186/s13059-020-02020-4.
55. Sankaranarayanan, K et al., <a href="https://doi.org/10.1016/j.cub.2015.10.060">"Gut Microbiome Diversity among Cheyenne and Arapaho Individuals from Western Oklahoma"</a>. Curr Biol. 2015 Dec 21;25(24):3161-9. doi: 10.1016/j.cub.2015.10.060. Epub 2015 Dec 6.
56. Schirmer, M et al., <a href="https://doi.org/10.1016/j.cell.2016.11.046">"Linking the Human Gut Microbiome to Inflammatory Cytokine Production Capacity"</a>. Cell. 2016 Dec 15;167(7):1897. doi: 10.1016/j.cell.2016.11.046.
57. Shao, Y et al., <a href="https://doi.org/10.1038/s41586-019-1560-1">"Stunted microbiota and opportunistic pathogen colonization in caesarean-section birth"</a>. Nature. 2019 Oct;574(7776):117-121. doi: 10.1038/s41586-019-1560-1. Epub 2019 Sep 18.
58. Smits, SA et al., <a href="https://doi.org/10.1126/science.aan4834">"Seasonal cycling in the gut microbiome of the Hadza hunter-gatherers of Tanzania"</a>. Science. 2017 Aug 25;357(6353):802-806. doi: 10.1126/science.aan4834.
59. Tett, A et al., <a href="https://doi.org/10.1016/j.chom.2019.08.018">"The Prevotella copri Complex Comprises Four Distinct Clades Underrepresented in Westernized Populations"</a>. Cell Host Microbe. 2019 Nov 13;26(5):666-679.e7. doi: 10.1016/j.chom.2019.08.018. Epub 2019 Oct 10.
60. Thomas, AM et al., <a href="https://doi.org/10.1038/s41591-019-0405-7">"Metagenomic analysis of colorectal cancer datasets identifies cross-cohort microbial diagnostic signatures and a link with choline degradation"</a>. Nat Med. 2019 Apr;25(4):667-678. doi: 10.1038/s41591-019-0405-7. Epub 2019 Apr 1.
61. Vatanen, T et al., <a href="https://doi.org/10.1016/j.cell.2016.04.007">"Variation in Microbiome LPS Immunogenicity Contributes to Autoimmunity in Humans"</a>. Cell. 2016 May 5;165(4):842-53. doi: 10.1016/j.cell.2016.04.007. Epub 2016 Apr 28.
62. Vich, Vila, A et al., <a href="https://doi.org/10.1126/scitranslmed.aap8914">"Gut microbiota composition and functional changes in inflammatory bowel disease and irritable bowel syndrome"</a>. Sci Transl Med. 2018 Dec 19;10(472):eaap8914. doi: 10.1126/scitranslmed.aap8914.
63. Vincent, C et al., <a href="https://doi.org/10.1186/s40168-016-0156-3">"Bloom and bust: intestinal microbiota dynamics in response to hospital exposures and Clostridium difficile colonization or infection"</a>. Microbiome. 2016 Mar 14;4:12. doi: 10.1186/s40168-016-0156-3.
64. Vogtmann, E et al., <a href="https://doi.org/10.1371/journal.pone.0155362">"Colorectal Cancer and the Human Gut Microbiome: Reproducibility with Whole-Genome Shotgun Sequencing"</a>. PLoS One. 2016 May 12;11(5):e0155362. doi: 10.1371/journal.pone.0155362. eCollection 2016.
65. Wampach, L et al., <a href="https://doi.org/10.1038/s41467-018-07631-x">"Birth mode is associated with earliest strain-conferred gut microbiome functions and immunostimulatory potential"</a>. Nat Commun. 2018 Nov 30;9(1):5091. doi: 10.1038/s41467-018-07631-x.
66. Wirbel, J et al., <a href="https://doi.org/10.1038/s41591-019-0406-6">"Meta-analysis of fecal metagenomes reveals global microbial signatures that are specific for colorectal cancer"</a>. Nat Med. 2019 Apr;25(4):679-689. doi: 10.1038/s41591-019-0406-6. Epub 2019 Apr 1.
67. Xie, H et al., <a href="https://doi.org/10.1016/j.cels.2016.10.004">"Shotgun Metagenomics of 250 Adult Twins Reveals Genetic and Environmental Impacts on the Gut Microbiome"</a>. Cell Syst. 2016 Dec 21;3(6):572-584.e3. doi: 10.1016/j.cels.2016.10.004. Epub 2016 Nov 3.
68. Yachida, S et al., <a href="https://doi.org/10.1038/s41591-019-0458-7">"Metagenomic and metabolomic analyses reveal distinct stage-specific phenotypes of the gut microbiota in colorectal cancer"</a>. Nat Med. 2019 Jun;25(6):968-976. doi: 10.1038/s41591-019-0458-7. Epub 2019 Jun 6.
69. Yassour, M et al., <a href="https://doi.org/10.1186/s13073-016-0271-6">"Sub-clinical detection of gut microbial biomarkers of obesity and type 2 diabetes"</a>. Genome Med. 2016 Feb 17;8(1):17. doi: 10.1186/s13073-016-0271-6.
70. Yassour, M et al., <a href="https://doi.org/10.1016/j.chom.2018.06.007">"Strain-Level Analysis of Mother-to-Child Bacterial Transmission during the First Few Months of Life"</a>. Cell Host Microbe. 2018 Jul 11;24(1):146-154.e4. doi: 10.1016/j.chom.2018.06.007.
71. Ye, Z et al., <a href="https://doi.org/10.1186/s40168-018-0520-6">"A metagenomic study of the gut microbiome in Behcet's disease"</a>. Microbiome. 2018 Aug 4;6(1):135. doi: 10.1186/s40168-018-0520-6.
72. Yu, J et al., <a href="https://doi.org/10.1136/gutjnl-2015-309800">"Metagenomic analysis of faecal microbiome as a tool towards targeted non-invasive biomarkers for colorectal cancer"</a>. Gut. 2017 Jan;66(1):70-78. doi: 10.1136/gutjnl-2015-309800. Epub 2015 Sep 25.
73. Zeevi, D et al., <a href="https://doi.org/10.1016/j.cell.2015.11.001">"Personalized Nutrition by Prediction of Glycemic Responses"</a>. Cell. 2015 Nov 19;163(5):1079-1094. doi: 10.1016/j.cell.2015.11.001.
74. Zeller, G et al., <a href="https://doi.org/10.15252/msb.20145645">"Potential of fecal microbiota for early-stage detection of colorectal cancer"</a>. Mol Syst Biol. 2014 Nov 28;10(11):766. doi: 10.15252/msb.20145645.
75. Zhu, F et al., <a href="https://doi.org/10.1038/s41467-020-15457-9">"Metagenome-wide association of gut microbiomeĘfeatures for schizophrenia"</a>. Nat Commun. 2020 Mar 31;11(1):1612. doi: 10.1038/s41467-020-15457-9.
