---
layout: page
# title: Microbiome Study Explorer
# description: 
# feature_image: images/about.jpg
permalink: study-explorer
---

<h1>Interactive Fermentation Product Secretion Explorer</h1>

This figure permits an exploration of the distribution of our estimated fermentation product 
secretion from published studies of human microbiome composition. The top panel allows 
you to select between different published studies and filter on individuals within 
different age categories and health status. The grey histogram (right) illustrates
what fraction of the total microbial biomass is represented by the bacterial species 
we experimentally examined in this study.


The middle panel allows you to either 
i) directly adjust how much microbiota available carbohydrate reaches the lower intestine,
ii) directly adjust how much starch and fibers are consumed, or iii) select different 
reference diets as described in the main text of the paper. 


The bottom panel shows the distributions of expected microbial biomass and 
fermentation products. Additionally, we show the total daily energy content of all 
secreted fermentation products. 
<hr/>

<center>

{% include interactive_excretion.html %}


</center>



### Referenced Data Sets 
1. Asnicar, F. *et al.*  [Microbiome connections with host metabolism and habitual diet from 1,098 deeply phenotyped individuals](https://www.nature.com/articles/s41591-020-01183-8). Nat Med 27, 321–332 (2021). 
2. Rubel, M. A. *et al.*  [Lifestyle and the presence of helminths is associated with gut microbiome composition in Cameroonians](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02020-4). Genome Biology 21, 122 (2020).
3. Tett, A. *et al.* [The *Prevotella copri* Complex Comprises Four Distinct Clades Underrepresented in Westernized Populations](https://www.sciencedirect.com/science/article/pii/S1931312819304275). Cell Host & Microbe 26, 666-679.e7 (2019).
4. Shao, Y. *et al.* [Stunted microbiota and opportunistic pathogen colonization in caesarean-section birth](https://www.nature.com/articles/s41586-019-1560-1). Nature 574, 117–121 (2019).
5. Yachida, S. *et al.* [Metagenomic and metabolomic analyses reveal distinct stage-specific phenotypes of the gut microbiota in colorectal cancer](https://www.nature.com/articles/s41591-019-0458-7). Nat Med 25, 968–976 (2019).
6. Zhou, W. *et al.* [Longitudinal multi-omics of host–microbe dynamics in prediabetes](https://www.nature.com/articles/s41586-019-1236-x). Nature 569, 663–671 (2019).
7. Wirbel, J. *et al.* [Meta-analysis of fecal metagenomes reveals global microbial signatures that are specific for colorectal cancer](https://www.nature.com/articles/s41591-019-0406-6). Nat Med 25, 679–689 (2019).
8. Pasolli, E. *et al.* [Extensive Unexplored Human Microbiome Diversity Revealed by Over 150,000 Genomes from Metagenomes Spanning Age, Geography, and Lifestyle](https://www.sciencedirect.com/science/article/pii/S0092867419300017). Cell 176, 649-662.e20 (2019).
9. Vich Vila, A. *et al.* [Gut microbiota composition and functional changes in inflammatory bowel disease and irritable bowel syndrome](https://www.science.org/doi/10.1126/scitranslmed.aap8914). Science Translational Medicine 10, eaap8914 (2018).
10. Yassour, M. *et al.* [Strain-Level Analysis of Mother-to-Child Bacterial Transmission during the First Few Months of Life](https://www.sciencedirect.com/science/article/pii/S1931312818303196). Cell Host & Microbe 24, 146-154.e4 (2018).
11. Hall, A. B. *et al.* [A novel *Ruminococcus gnavus* clade enriched in inflammatory bowel disease patients](https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-017-0490-5). Genome Medicine 9, 103 (2017).
12. Zhernakova, A. *et al.* [Population-based metagenomics analysis reveals markers for gut microbiome composition and diversity](https://www.science.org/doi/10.1126/science.aad3369). Science 352, 565–569 (2016).
13. Zeller, G. *et al.* [Potential of fecal microbiota for early‐stage detection of colorectal cancer](https://www.embopress.org/doi/full/10.15252/msb.20145645). Molecular Systems Biology 10, 766 (2014).
14. Jie, Z. *et al.* [The gut microbiome in atherosclerotic cardiovascular disease](https://www.nature.com/articles/s41467-017-00900-1). Nat Commun 8, 845 (2017).
15. Brito, I. L. *et al.* [Mobile genes in the human microbiome are structured from global to individual scales](https://www.nature.com/articles/nature18927). Nature 535, 435–439 (2016).
16. Pehrsson, E. C. *et al.* [Interconnected microbiomes and resistomes in low-income human habitats](https://www.nature.com/articles/nature17672). Nature 533, 212–216 (2016).
17. Bäckhed, F. *et al.* [Dynamics and Stabilization of the Human Gut Microbiome during the First Year of Life](https://www.sciencedirect.com/science/article/pii/S1931312815001626). Cell Host & Microbe 17, 690–703 (2015).
18. Nielsen, H. B. *et al.* [Identification and assembly of genomes and genetic elements in complex metagenomic samples without using reference genomes](https://www.nature.com/articles/nbt.2939). Nat Biotechnol 32, 822–828 (2014).
19. Li, J. *et al.* [An integrated catalog of reference genes in the human gut microbiome](https://www.nature.com/articles/nbt.2942). Nat Biotechnol 32, 834–841 (2014).
20. Huttenhower, C. *et al.* [Structure, function and diversity of the healthy human microbiome](https://www.nature.com/articles/nature11234). Nature 486, 207–214 (2012).

### Reference Taxonomic Studies 
Analysis based on cross-study taxonomic abundances reported in the Curated Metagenomic Data project (Pasolli *et al.*  [Accessible, curated metagenomic data through ExperimentHub](https://www.nature.com/articles/nmeth.4468). Nat. Methods. (2017).)