{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plotting of experimental data\n",
    "\n",
    "- 2a: plotting of distribution\n",
    "- "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>species_HPLCname</th>\n",
       "      <th>species</th>\n",
       "      <th>new_species</th>\n",
       "      <th>species.1</th>\n",
       "      <th>species_short</th>\n",
       "      <th>new_genus</th>\n",
       "      <th>genus</th>\n",
       "      <th>new_family</th>\n",
       "      <th>family</th>\n",
       "      <th>new_order</th>\n",
       "      <th>order</th>\n",
       "      <th>new_class</th>\n",
       "      <th>class</th>\n",
       "      <th>new_phylum</th>\n",
       "      <th>phylum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B.vulgatus</td>\n",
       "      <td>Phocaeicola vulgatus</td>\n",
       "      <td>Phocaeicola vulgatus</td>\n",
       "      <td>Bacteroides vulgatus</td>\n",
       "      <td>B. vulgatus</td>\n",
       "      <td>Phocaeicola</td>\n",
       "      <td>Phocaeicola</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B.fragilis</td>\n",
       "      <td>Bacteroides fragilis</td>\n",
       "      <td>Bacteroides fragilis</td>\n",
       "      <td>Bacteroides fragilis</td>\n",
       "      <td>B. fragilis</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>B.ovatus</td>\n",
       "      <td>Bacteroides ovatus</td>\n",
       "      <td>Bacteroides ovatus</td>\n",
       "      <td>Bacteroides ovatus</td>\n",
       "      <td>B. ovatus</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B.theta</td>\n",
       "      <td>Bacteroides thetaiotaomicron</td>\n",
       "      <td>Bacteroides thetaiotaomicron</td>\n",
       "      <td>Bacteroides thetaiotaomicron</td>\n",
       "      <td>B. theta</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P.copri</td>\n",
       "      <td>Prevotella copri</td>\n",
       "      <td>Prevotella copri</td>\n",
       "      <td>Prevotella copri</td>\n",
       "      <td>P. copri</td>\n",
       "      <td>Prevotella</td>\n",
       "      <td>Prevotella</td>\n",
       "      <td>Prevotellaceae</td>\n",
       "      <td>Prevotellaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  species_HPLCname                       species  \\\n",
       "0       B.vulgatus          Phocaeicola vulgatus   \n",
       "1       B.fragilis          Bacteroides fragilis   \n",
       "2         B.ovatus            Bacteroides ovatus   \n",
       "3          B.theta  Bacteroides thetaiotaomicron   \n",
       "4          P.copri              Prevotella copri   \n",
       "\n",
       "                    new_species                     species.1 species_short  \\\n",
       "0          Phocaeicola vulgatus          Bacteroides vulgatus   B. vulgatus   \n",
       "1          Bacteroides fragilis          Bacteroides fragilis   B. fragilis   \n",
       "2            Bacteroides ovatus            Bacteroides ovatus     B. ovatus   \n",
       "3  Bacteroides thetaiotaomicron  Bacteroides thetaiotaomicron      B. theta   \n",
       "4              Prevotella copri              Prevotella copri      P. copri   \n",
       "\n",
       "     new_genus        genus      new_family          family      new_order  \\\n",
       "0  Phocaeicola  Phocaeicola  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "1  Bacteroides  Bacteroides  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "2  Bacteroides  Bacteroides  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "3  Bacteroides  Bacteroides  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "4   Prevotella   Prevotella  Prevotellaceae  Prevotellaceae  Bacteroidales   \n",
       "\n",
       "           order    new_class        class    new_phylum         phylum  \n",
       "0  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "1  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "2  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "3  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "4  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import sys, os\n",
    "import glob \n",
    "import matplotlib as mpl\n",
    "mpl.rcParams['pdf.fonttype'] = 42\n",
    "import collections\n",
    "import builtins\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import scipy.integrate as spi\n",
    "from scipy.integrate import odeint #this is the module to solve ODEs\n",
    "\n",
    "import met_brewer\n",
    "\n",
    "%matplotlib inline\n",
    "# Load the scipy module/\n",
    "import scipy.stats \n",
    "import json\n",
    "\n",
    "import csv\n",
    "from collections import defaultdict\n",
    "from pprint import pprint\n",
    "\n",
    "################################\n",
    "#load information of species to include\n",
    "#################################\n",
    "#load species information for all characterized species\n",
    "speciesinformation=pd.read_csv(\"data_hplc/species_properties.csv\",skiprows=1)\n",
    "display(speciesinformation.head())\n",
    "\n",
    "\n",
    "from ete3 import Tree, TreeStyle, TextFace, add_face_to_node\n",
    "import plot_eteTree \n",
    "#adjusted from https://gist.github.com/jolespin/5d90deff552138d73de7ed4bdd9ac57a\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load general files and generate tree\n",
    "\n",
    "## Note: This uses the taxonomic abundance tables caliulatioed in subfolder microbiota_composition. To decide if all or only healthy samples should be considered run the script in the folder first and make your selection there.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Phocaeicola vulgatus', 'Bacteroides fragilis', 'Bacteroides ovatus', 'Bacteroides thetaiotaomicron', 'Bacteroides finegoldii', 'Bacteroides uniformis', 'Prevotella copri', 'Parabacteroides distasonis', 'Eubacterium rectale', 'Roseburia intestinalis', 'Faecalibacterium prausnitzii', 'Ruminococcus bromii', 'Bifidobacterium longum', 'Bifidobacterium adolescentis', 'Collinsella aerofaciens', 'Escherichia coli']\n",
      "['Bacteroidetes', 'Firmicutes', 'Actinobacteria', 'Proteobacteria', nan]\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "nan",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m~/mambaforge/envs/ete/lib/python3.9/site-packages/pandas/core/indexes/base.py:3653\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3652\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3653\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3654\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m~/mambaforge/envs/ete/lib/python3.9/site-packages/pandas/_libs/index.pyx:147\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/mambaforge/envs/ete/lib/python3.9/site-packages/pandas/_libs/index.pyx:176\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:7080\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:7088\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: nan",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 133\u001b[0m\n\u001b[1;32m    130\u001b[0m     \u001b[38;5;28mprint\u001b[39m(taxa_levellist[lc])\n\u001b[1;32m    131\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m taxidentifier \u001b[38;5;129;01min\u001b[39;00m taxa_levellist[lc]:\n\u001b[1;32m    132\u001b[0m         \u001b[38;5;66;03m#print(taxidentifier)\u001b[39;00m\n\u001b[0;32m--> 133\u001b[0m         abundancevalues[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]\u001b[38;5;241m.\u001b[39mappend(\u001b[43mabundanceinformation\u001b[49m\u001b[43m[\u001b[49m\u001b[43mlc\u001b[49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mat\u001b[49m\u001b[43m[\u001b[49m\u001b[43mtaxidentifier\u001b[49m\u001b[43m,\u001b[49m\u001b[43mstudy\u001b[49m\u001b[43m]\u001b[49m)\n\u001b[1;32m    136\u001b[0m \u001b[38;5;66;03m#generate list with abundance for all taxonomic levels\u001b[39;00m\n\u001b[1;32m    137\u001b[0m alltaxonomicnames\u001b[38;5;241m=\u001b[39m[]\n",
      "File \u001b[0;32m~/mambaforge/envs/ete/lib/python3.9/site-packages/pandas/core/indexing.py:2419\u001b[0m, in \u001b[0;36m_AtIndexer.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2416\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInvalid call for scalar access (getting)!\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   2417\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mobj\u001b[38;5;241m.\u001b[39mloc[key]\n\u001b[0;32m-> 2419\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43msuper\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__getitem__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/mambaforge/envs/ete/lib/python3.9/site-packages/pandas/core/indexing.py:2371\u001b[0m, in \u001b[0;36m_ScalarAccessIndexer.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2368\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mInvalid call for scalar access (getting)!\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m   2370\u001b[0m key \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_convert_key(key)\n\u001b[0;32m-> 2371\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mobj\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_value\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtakeable\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_takeable\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/mambaforge/envs/ete/lib/python3.9/site-packages/pandas/core/frame.py:3877\u001b[0m, in \u001b[0;36mDataFrame._get_value\u001b[0;34m(self, index, col, takeable)\u001b[0m\n\u001b[1;32m   3871\u001b[0m engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39m_engine\n\u001b[1;32m   3873\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex, MultiIndex):\n\u001b[1;32m   3874\u001b[0m     \u001b[38;5;66;03m# CategoricalIndex: Trying to use the engine fastpath may give incorrect\u001b[39;00m\n\u001b[1;32m   3875\u001b[0m     \u001b[38;5;66;03m#  results if our categories are integers that dont match our codes\u001b[39;00m\n\u001b[1;32m   3876\u001b[0m     \u001b[38;5;66;03m# IntervalIndex: IntervalTree has no get_loc\u001b[39;00m\n\u001b[0;32m-> 3877\u001b[0m     row \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindex\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3878\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m series\u001b[38;5;241m.\u001b[39m_values[row]\n\u001b[1;32m   3880\u001b[0m \u001b[38;5;66;03m# For MultiIndex going through engine effectively restricts us to\u001b[39;00m\n\u001b[1;32m   3881\u001b[0m \u001b[38;5;66;03m#  same-length tuples; see test_get_set_value_no_partial_indexing\u001b[39;00m\n",
      "File \u001b[0;32m~/mambaforge/envs/ete/lib/python3.9/site-packages/pandas/core/indexes/base.py:3655\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3653\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3654\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m-> 3655\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3656\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3657\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3658\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3659\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3660\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: nan"
     ]
    }
   ],
   "source": [
    "############################\n",
    "#plotting options for tree\n",
    "##########################\n",
    "phyla_colors=[\"r\",\"b\",\"m\",\"orange\",\"brown\"] \n",
    "phyla=['Bacteroidetes', 'Firmicutes', 'Actinobacteria', 'Proteobacteria']\n",
    "\n",
    "#define which study to use for abundance data\n",
    "study=\"HMP_2019_ibdmdb\"\n",
    "\n",
    "\n",
    "\n",
    "################################\n",
    "#load abundance information (from analysis of metagenomics data\"\n",
    "# ! modify here if other studies should be included\n",
    "################################\n",
    "abundanceinformation=[]\n",
    "taxa_levellist=[]\n",
    "taxa_levellistC=[]\n",
    "for level in [\"phylum\",\"class\",\"order\",\"family\",\"genus\",\"species\"]:\n",
    "    taxa_levellist.append([])\n",
    "    taxa_levellistC.append([])\n",
    "    abundanceinformation.append(pd.read_csv(\"../microbiota_composition/avabundance_\"+level+\".csv\",skiprows=0,index_col=0,))\n",
    "\n",
    "all_species_list=[]\n",
    "all_species_list_short=[]\n",
    "#################################\n",
    "#generate Newick tree\n",
    "################################\n",
    "branchdistance=\":2\" #distance between branches\n",
    "branchdistance_species=\":0.3\" #distance of lead (last name)\n",
    "\n",
    "#start analysis\n",
    "branchdistance_number=float(branchdistance[1:])\n",
    "phyla=speciesinformation[\"phylum\"].unique()\n",
    "newick=\"(\"\n",
    "cP=-1\n",
    "\n",
    "#go through each taxonomic level\n",
    "for phylum in phyla:\n",
    "    taxa_levellist[0].append(phylum)  \n",
    "    cP=cP+1\n",
    "    taxa_levellistC[0].append(phyla_colors[cP])\n",
    "    select=speciesinformation.loc[speciesinformation[\"phylum\"]==phylum]\n",
    "    classes=select[\"class\"].unique()\n",
    "    \n",
    "    if cP==0:\n",
    "            newick=newick+\"(\"\n",
    "    else:\n",
    "            newick=newick+\",(\"\n",
    "    cI=-1    \n",
    "    for classc in classes:\n",
    "        taxa_levellist[1].append(classc)\n",
    "        taxa_levellistC[1].append(phyla_colors[cP])\n",
    "        cI=cI+1\n",
    "        select1=select.loc[select[\"class\"]==classc]\n",
    "        if cI==0:\n",
    "            newick=newick+\"(\"\n",
    "        else:\n",
    "            newick=newick+\",(\"\n",
    "        orders=select1[\"order\"].unique()\n",
    "        oI= -1\n",
    "        for order in orders:\n",
    "            taxa_levellist[2].append(order)\n",
    "            taxa_levellistC[2].append(phyla_colors[cP])\n",
    "            oI=oI+1\n",
    "            select2=select1.loc[select1[\"order\"]==order]\n",
    "            if oI==0:\n",
    "                newick=newick+\"(\"\n",
    "            else:\n",
    "                newick=newick+\",(\"\n",
    "            families=select2[\"family\"].unique()\n",
    "            fI=-1\n",
    "            for family in families:\n",
    "                taxa_levellist[3].append(family)\n",
    "                taxa_levellistC[3].append(phyla_colors[cP])\n",
    "                fI=fI+1\n",
    "                select3=select2.loc[select2[\"family\"]==family]\n",
    "                genus=select3[\"genus\"].unique()\n",
    "                if fI==0:\n",
    "                    newick=newick+\"(\"\n",
    "                else:\n",
    "                    newick=newick+\",(\"\n",
    "\n",
    "                gI=-1\n",
    "                for genus in genus:\n",
    "                    taxa_levellist[4].append(genus)\n",
    "                    taxa_levellistC[4].append(phyla_colors[cP])\n",
    "                    gI=gI+1\n",
    "                    select4=select3.loc[select3[\"genus\"]==genus]\n",
    "                    species=select4[\"species\"].unique()\n",
    "                    if gI==0:\n",
    "                        newick=newick+\"(\"\n",
    "                    else:\n",
    "                        newick=newick+\",(\"\n",
    "                    spc=-1\n",
    "                    for speciesc in species:\n",
    "                            taxa_levellist[5].append(speciesc)\n",
    "                            taxa_levellistC[5].append(phyla_colors[cP])\n",
    "                            spc=spc+1\n",
    "                            if spc>0:\n",
    "                                newick=newick+\",\"\n",
    "                            select5=select4.loc[select4[\"species\"]==speciesc]\n",
    "                            all_species_list.append(speciesc)\n",
    "                            newick= newick+speciesc+branchdistance_species\n",
    "                    newick=newick+\")\"+genus+branchdistance\n",
    "                newick=newick+\")\"+family+branchdistance\n",
    "            newick=newick+\")\"+order+branchdistance\n",
    "        newick=newick+\")\"+classc+branchdistance\n",
    "    newick=newick+\")\"+str(phylum)+branchdistance  \n",
    "newick=newick+\");\"\n",
    "\n",
    "#save tree\n",
    "text_file = open(\"newick.txt\", \"w\")\n",
    "n = text_file.write(newick)\n",
    "text_file.close()\n",
    "\n",
    "print(all_species_list)\n",
    "\n",
    "\n",
    "#####################\n",
    "#for different nodes of tree get abundance lvele\n",
    "######################\n",
    "    \n",
    "\n",
    "lc=-1\n",
    "abundancevalues=[]\n",
    "for level in [\"phylum\",\"class\",\"order\",\"family\",\"genus\",\"species\"]:\n",
    "    abundancevalues.append([])\n",
    "    lc=lc+1\n",
    "    print(taxa_levellist[lc])\n",
    "    for taxidentifier in taxa_levellist[lc]:\n",
    "        #print(taxidentifier)\n",
    "        abundancevalues[-1].append(abundanceinformation[lc].at[taxidentifier,study])\n",
    "\n",
    "\n",
    "#generate list with abundance for all taxonomic levels\n",
    "alltaxonomicnames=[]\n",
    "alltaxonomiccolors=[] #color (currently based on phyla)\n",
    "alltaxonomiccolors_phylum=[] #color (phyla in color rest black)\n",
    "taxonomiclevel=[]\n",
    "\n",
    "alltaxonomicabundances=[]\n",
    "abundancevalues_sum=[]\n",
    "lc=-1\n",
    "for level in [\"phylum\",\"class\",\"order\",\"family\",\"genus\",\"species\"]:\n",
    "    lc=lc+1\n",
    "    #colors for taxonomic names\n",
    "    #option 1: only phylum in phylum color, others black\n",
    "    #if level==\"phylum\":\n",
    "    #    alltaxonomiccolors_phylum=alltaxonomiccolors_phylum+taxa_levellistC[lc]\n",
    "    #else:\n",
    "    #    alltaxonomiccolors_phylum=alltaxonomiccolors_phylum+[\"k\"]*len(taxa_levellist[lc])\n",
    "    #option 2: set all names according to phylum:\n",
    "    alltaxonomiccolors_phylum=alltaxonomiccolors_phylum+taxa_levellistC[lc]\n",
    "    taxonomiclevel=taxonomiclevel+[level]*len(taxa_levellist[lc])\n",
    "\n",
    "   \n",
    "    alltaxonomicnames=alltaxonomicnames+taxa_levellist[lc]\n",
    "    alltaxonomiccolors=alltaxonomiccolors+taxa_levellistC[lc]\n",
    "    abundancevalues_sum.append(sum(abundancevalues[lc]))\n",
    "    alltaxonomicabundances=alltaxonomicabundances+abundancevalues[lc]\n",
    "\n",
    "#for easier handling, generate dataframe with all information\n",
    "data_abundancec=pd.DataFrame(list(zip(alltaxonomicnames, alltaxonomicabundances, alltaxonomiccolors,alltaxonomiccolors_phylum,taxonomiclevel)))\n",
    "data_abundancec.columns=[\"taxidentifier\",\"abundance\",\"color\",\"color_phylum\",\"tax_level\"]\n",
    "data_abundancec.set_index(\"taxidentifier\",inplace=True)\n",
    "print(\"sum of abundance\")\n",
    "print(abundancevalues_sum)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#add short and alternamtive names according to \n",
    "for iter, row in data_abundancec.iterrows():\n",
    "    levelc=row[\"tax_level\"]\n",
    "    taxind=iter\n",
    "    curinfo=speciesinformation.loc[speciesinformation[levelc]==iter]\n",
    "    if levelc==\"species\":\n",
    "        shortn=curinfo[level+\"_short\"]\n",
    "        data_abundancec.at[iter,\"short_name\"]=shortn.iloc[0]\n",
    "    shortn=curinfo[\"new_\"+levelc]\n",
    "    if (shortn.shape[0]>1) and (levelc==\"species\"):\n",
    "        error\n",
    "    data_abundancec.at[iter,\"new_name\"]=shortn.iloc[0]\n",
    "\n",
    "display(data_abundancec.head(15))\n",
    "#consistency check, taxonomic identifiers should be unique\n",
    "if len(alltaxonomicnames) != len(set(alltaxonomicnames)):\n",
    "    print(\"Error: non unique used of taxonomic identifiers\")\n",
    "    print([item for item, count in collections.Counter(alltaxonomicnames).items() if count > 1])\n",
    "    error\n",
    "\n",
    "############\n",
    "#generate ete3 tree from Newick tree\n",
    "############\n",
    "\n",
    "t = Tree(newick, format=1)\n",
    "ts = TreeStyle()\n",
    "ts.show_leaf_name = False\n",
    "\n",
    "############\n",
    "#ad properties to tree branches\n",
    "#############\n",
    "\n",
    "all_species_list_short=[]\n",
    "for leaf in t.traverse():\n",
    "    nameleaf=leaf.name\n",
    "    if len(nameleaf)>0:\n",
    "        abundancec=str(round(data_abundancec.at[nameleaf,\"abundance\"],1))+\"%\"\n",
    "\n",
    "        \n",
    "        color=data_abundancec.at[nameleaf,\"color\"] #color for lines\n",
    "        color_phylum=data_abundancec.at[nameleaf,\"color_phylum\"] #color for taxonomic names\n",
    "        new_name=data_abundancec.at[nameleaf,\"new_name\"]\n",
    "        short_name=data_abundancec.at[nameleaf,\"short_name\"]\n",
    "        if short_name != \"nan\":\n",
    "            all_species_list_short.append(short_name)\n",
    "        #print(nameleaf)\n",
    "        #print(abundancec)\n",
    "        leaf.add_features(abundance=abundancec)\n",
    "        leaf.add_features(new_name=new_name)\n",
    "        leaf.add_features(short_name=short_name)\n",
    "        leaf.add_features(color=color_phylum)\n",
    "        leaf.img_style['hz_line_color'] = color\n",
    "        leaf.img_style['vt_line_color'] = color\n",
    "    else:\n",
    "        leaf.add_features(abundance=\"\")\n",
    "        leaf.add_features(abundance=\"\")\n",
    "        leaf.add_features(new_name=\"\")\n",
    "        leaf.add_features(short_name=\"\")\n",
    "        leaf.add_features(color=\"k\")\n",
    "        leaf.img_style['hz_line_color'] = \"k\"\n",
    "        leaf.img_style['vt_line_color'] = \"k\"\n",
    "\n",
    "        \n",
    "    leaf.img_style['hz_line_width']=2\n",
    "    leaf.img_style['vt_line_width']=2\n",
    "    leaf.img_style['fgcolor']='#000000'\n",
    "\n",
    "print(all_species_list_short)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plot tree\n",
    "## !! This usese the ete enviornment. Activate enviornment first (or create an enviornment and install ete)\n",
    "\n",
    "When I run it there was a problem with Python >3.9 so I installed it using Pytohn 3.9\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plot tree with full species information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 't' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[9], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m fig, ax \u001b[38;5;241m=\u001b[39m plt\u001b[38;5;241m.\u001b[39msubplots(\u001b[38;5;241m1\u001b[39m,\u001b[38;5;241m1\u001b[39m,figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m12\u001b[39m,\u001b[38;5;241m5\u001b[39m))  \u001b[38;5;66;03m#for barplots yields/excretion\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m plot_eteTree\u001b[38;5;241m.\u001b[39mplot_tree(\u001b[43mt\u001b[49m, align_names\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, name_offset\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, max_dist\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, font_size\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m9\u001b[39m, axe\u001b[38;5;241m=\u001b[39max,font_size2\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m8\u001b[39m,alternative_names\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,short_names\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n\u001b[1;32m      3\u001b[0m lc\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m level \u001b[38;5;129;01min\u001b[39;00m [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mphylum\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mclass\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124morder\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfamily\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgenus\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mspecies\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n",
      "\u001b[0;31mNameError\u001b[0m: name 't' is not defined"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA+AAAAGyCAYAAABk/q6oAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAf30lEQVR4nO3df2zV9b348VdpaavutoswaxHs6qaTXTJ2aQOj3GbRaQ0Ybkh2QxcXq15M1my7BHp1A1l0EJNmu5m51ym4RdAsQW/jz/hH52huNn4IS0ZTlkXI3SJcC1srac1a1N0i8Ll/+KXf27Uo59C+aeXxSM4ffe/9Pud9lvfqnn7O6acgy7IsAAAAgAk17WJvAAAAAC4FAhwAAAASEOAAAACQgAAHAACABAQ4AAAAJCDAAQAAIAEBDgAAAAkIcAAAAEhAgAMAAEACAhwAAAASyDnAd+3aFcuXL49Zs2ZFQUFBvPzyyx+5ZufOnVFTUxOlpaVx3XXXxRNPPJHPXgEAAGDKyjnA33333Zg/f3489thj5zX/yJEjsWzZsqivr4+urq544IEHYvXq1fHCCy/kvFkAAACYqgqyLMvyXlxQEC+99FKsWLHinHO++93vxiuvvBKHDh0aHmtubo7f/va3sW/fvnxfGgAAAKaUool+gX379kVDQ8OIsdtuuy22bt0a77//fkyfPn3UmqGhoRgaGhr++cyZM/H222/HjBkzoqCgYKK3DAAAwCUuy7I4ceJEzJo1K6ZNG58/nzbhAd7b2xsVFRUjxioqKuLUqVPR19cXlZWVo9a0trbGxo0bJ3prAAAA8KGOHj0as2fPHpfnmvAAj4hRV63Pfur9XFez169fHy0tLcM/DwwMxLXXXhtHjx6NsrKyidsoAAAARMTg4GDMmTMn/uZv/mbcnnPCA/zqq6+O3t7eEWPHjx+PoqKimDFjxphrSkpKoqSkZNR4WVmZAAcAACCZ8fwa9ITfB3zx4sXR0dExYmzHjh1RW1s75ve/AQAA4OMo5wB/55134sCBA3HgwIGI+OA2YwcOHIju7u6I+ODj401NTcPzm5ub480334yWlpY4dOhQbNu2LbZu3Rr33Xff+LwDAAAAmAJy/gj6/v3746abbhr++ex3te+66654+umno6enZzjGIyKqq6ujvb091q5dG48//njMmjUrHn300fjqV786DtsHAACAqeGC7gOeyuDgYJSXl8fAwIDvgAMAADDhJqJDJ/w74AAAAIAABwAAgCQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgATyCvDNmzdHdXV1lJaWRk1NTezevftD52/fvj3mz58fl19+eVRWVsY999wT/f39eW0YAAAApqKcA7ytrS3WrFkTGzZsiK6urqivr4+lS5dGd3f3mPP37NkTTU1NsWrVqnj99dfjueeei9/85jdx7733XvDmAQAAYKrIOcAfeeSRWLVqVdx7770xd+7c+Ld/+7eYM2dObNmyZcz5v/71r+PTn/50rF69Oqqrq+Pv//7v4xvf+Ebs37//gjcPAAAAU0VOAX7y5Mno7OyMhoaGEeMNDQ2xd+/eMdfU1dXFsWPHor29PbIsi7feeiuef/75uP3228/5OkNDQzE4ODjiAQAAAFNZTgHe19cXp0+fjoqKihHjFRUV0dvbO+aaurq62L59ezQ2NkZxcXFcffXV8clPfjJ+/OMfn/N1Wltbo7y8fPgxZ86cXLYJAAAAk05ef4StoKBgxM9Zlo0aO+vgwYOxevXqePDBB6OzszNeffXVOHLkSDQ3N5/z+devXx8DAwPDj6NHj+azTQAAAJg0inKZPHPmzCgsLBx1tfv48eOjroqf1draGkuWLIn7778/IiK+8IUvxBVXXBH19fXx8MMPR2Vl5ag1JSUlUVJSksvWAAAAYFLL6Qp4cXFx1NTUREdHx4jxjo6OqKurG3PNe++9F9OmjXyZwsLCiPjgyjkAAABcCnL+CHpLS0s8+eSTsW3btjh06FCsXbs2uru7hz9Svn79+mhqahqev3z58njxxRdjy5Ytcfjw4Xjttddi9erVsXDhwpg1a9b4vRMAAACYxHL6CHpERGNjY/T398emTZuip6cn5s2bF+3t7VFVVRURET09PSPuCX733XfHiRMn4rHHHot/+Zd/iU9+8pNx8803xw9+8IPxexcAAAAwyRVkU+Bz4IODg1FeXh4DAwNRVlZ2sbcDAADAx9xEdGhefwUdAAAAyI0ABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgATyCvDNmzdHdXV1lJaWRk1NTezevftD5w8NDcWGDRuiqqoqSkpK4jOf+Uxs27Ytrw0DAADAVFSU64K2trZYs2ZNbN68OZYsWRI/+clPYunSpXHw4MG49tprx1yzcuXKeOutt2Lr1q3x2c9+No4fPx6nTp264M0DAADAVFGQZVmWy4JFixbFggULYsuWLcNjc+fOjRUrVkRra+uo+a+++mp87Wtfi8OHD8eVV16Z1yYHBwejvLw8BgYGoqysLK/nAAAAgPM1ER2a00fQT548GZ2dndHQ0DBivKGhIfbu3TvmmldeeSVqa2vjhz/8YVxzzTVxww03xH333Rd/+ctfzvk6Q0NDMTg4OOIBAAAAU1lOH0Hv6+uL06dPR0VFxYjxioqK6O3tHXPN4cOHY8+ePVFaWhovvfRS9PX1xTe/+c14++23z/k98NbW1ti4cWMuWwMAAIBJLa8/wlZQUDDi5yzLRo2ddebMmSgoKIjt27fHwoULY9myZfHII4/E008/fc6r4OvXr4+BgYHhx9GjR/PZJgAAAEwaOV0BnzlzZhQWFo662n38+PFRV8XPqqysjGuuuSbKy8uHx+bOnRtZlsWxY8fi+uuvH7WmpKQkSkpKctkaAAAATGo5XQEvLi6Ompqa6OjoGDHe0dERdXV1Y65ZsmRJ/OlPf4p33nlneOz3v/99TJs2LWbPnp3HlgEAAGDqyfkj6C0tLfHkk0/Gtm3b4tChQ7F27dro7u6O5ubmiPjg4+NNTU3D8++4446YMWNG3HPPPXHw4MHYtWtX3H///fFP//RPcdlll43fOwEAAIBJLOf7gDc2NkZ/f39s2rQpenp6Yt68edHe3h5VVVUREdHT0xPd3d3D8z/xiU9ER0dH/PM//3PU1tbGjBkzYuXKlfHwww+P37sAAACASS7n+4BfDO4DDgAAQEoX/T7gAAAAQH4EOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACSQV4Bv3rw5qquro7S0NGpqamL37t3nte61116LoqKi+OIXv5jPywIAAMCUlXOAt7W1xZo1a2LDhg3R1dUV9fX1sXTp0uju7v7QdQMDA9HU1BRf+cpX8t4sAAAATFUFWZZluSxYtGhRLFiwILZs2TI8Nnfu3FixYkW0traec93Xvva1uP7666OwsDBefvnlOHDgwHm/5uDgYJSXl8fAwECUlZXlsl0AAADI2UR0aE5XwE+ePBmdnZ3R0NAwYryhoSH27t17znVPPfVUvPHGG/HQQw+d1+sMDQ3F4ODgiAcAAABMZTkFeF9fX5w+fToqKipGjFdUVERvb++Ya/7whz/EunXrYvv27VFUVHRer9Pa2hrl5eXDjzlz5uSyTQAAAJh08vojbAUFBSN+zrJs1FhExOnTp+OOO+6IjRs3xg033HDez79+/foYGBgYfhw9ejSfbQIAAMCkcX6XpP+fmTNnRmFh4air3cePHx91VTwi4sSJE7F///7o6uqKb3/72xERcebMmciyLIqKimLHjh1x8803j1pXUlISJSUluWwNAAAAJrWcroAXFxdHTU1NdHR0jBjv6OiIurq6UfPLysrid7/7XRw4cGD40dzcHJ/73OfiwIEDsWjRogvbPQAAAEwROV0Bj4hoaWmJO++8M2pra2Px4sXx05/+NLq7u6O5uTkiPvj4+B//+Mf42c9+FtOmTYt58+aNWH/VVVdFaWnpqHEAAAD4OMs5wBsbG6O/vz82bdoUPT09MW/evGhvb4+qqqqIiOjp6fnIe4IDAADApSbn+4BfDO4DDgAAQEoX/T7gAAAAQH4EOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACSQV4Bv3rw5qquro7S0NGpqamL37t3nnPviiy/GrbfeGp/61KeirKwsFi9eHL/4xS/y3jAAAABMRTkHeFtbW6xZsyY2bNgQXV1dUV9fH0uXLo3u7u4x5+/atStuvfXWaG9vj87Ozrjpppti+fLl0dXVdcGbBwAAgKmiIMuyLJcFixYtigULFsSWLVuGx+bOnRsrVqyI1tbW83qOv/3bv43GxsZ48MEHz2v+4OBglJeXx8DAQJSVleWyXQAAAMjZRHRoTlfAT548GZ2dndHQ0DBivKGhIfbu3Xtez3HmzJk4ceJEXHnlleecMzQ0FIODgyMeAAAAMJXlFOB9fX1x+vTpqKioGDFeUVERvb295/UcP/rRj+Ldd9+NlStXnnNOa2trlJeXDz/mzJmTyzYBAABg0snrj7AVFBSM+DnLslFjY3n22Wfj+9//frS1tcVVV111znnr16+PgYGB4cfRo0fz2SYAAABMGkW5TJ45c2YUFhaOutp9/PjxUVfF/1pbW1usWrUqnnvuubjllls+dG5JSUmUlJTksjUAAACY1HK6Al5cXBw1NTXR0dExYryjoyPq6urOue7ZZ5+Nu+++O5555pm4/fbb89spAAAATGE5XQGPiGhpaYk777wzamtrY/HixfHTn/40uru7o7m5OSI++Pj4H//4x/jZz34WER/Ed1NTU/z7v/97fOlLXxq+en7ZZZdFeXn5OL4VAAAAmLxyDvDGxsbo7++PTZs2RU9PT8ybNy/a29ujqqoqIiJ6enpG3BP8Jz/5SZw6dSq+9a1vxbe+9a3h8bvuuiuefvrpC38HAAAAMAXkfB/wi8F9wAEAAEjpot8HHAAAAMiPAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAEBDgAAAAkIMABAAAgAQEOAAAACQhwAAAASECAAwAAQAICHAAAABIQ4AAAAJCAAAcAAIAE8grwzZs3R3V1dZSWlkZNTU3s3r37Q+fv3LkzampqorS0NK677rp44okn8tosAAAATFU5B3hbW1usWbMmNmzYEF1dXVFfXx9Lly6N7u7uMecfOXIkli1bFvX19dHV1RUPPPBArF69Ol544YUL3jwAAABMFQVZlmW5LFi0aFEsWLAgtmzZMjw2d+7cWLFiRbS2to6a/93vfjdeeeWVOHTo0PBYc3Nz/Pa3v419+/ad12sODg5GeXl5DAwMRFlZWS7bBQAAgJxNRIcW5TL55MmT0dnZGevWrRsx3tDQEHv37h1zzb59+6KhoWHE2G233RZbt26N999/P6ZPnz5qzdDQUAwNDQ3/PDAwEBEf/BcAAAAAE+1sf+Z4zfpD5RTgfX19cfr06aioqBgxXlFREb29vWOu6e3tHXP+qVOnoq+vLyorK0etaW1tjY0bN44anzNnTi7bBQAAgAvS398f5eXl4/JcOQX4WQUFBSN+zrJs1NhHzR9r/Kz169dHS0vL8M9//vOfo6qqKrq7u8ftjcNkMzg4GHPmzImjR4/6qgUfW845lwLnnEuBc86lYGBgIK699tq48sorx+05cwrwmTNnRmFh4air3cePHx91lfusq6++esz5RUVFMWPGjDHXlJSURElJyajx8vJy/wPnY6+srMw552PPOedS4JxzKXDOuRRMmzZ+d+/O6ZmKi4ujpqYmOjo6Rox3dHREXV3dmGsWL148av6OHTuitrZ2zO9/AwAAwMdRzinf0tISTz75ZGzbti0OHToUa9euje7u7mhubo6IDz4+3tTUNDy/ubk53nzzzWhpaYlDhw7Ftm3bYuvWrXHfffeN37sAAACASS7n74A3NjZGf39/bNq0KXp6emLevHnR3t4eVVVVERHR09Mz4p7g1dXV0d7eHmvXro3HH388Zs2aFY8++mh89atfPe/XLCkpiYceemjMj6XDx4VzzqXAOedS4JxzKXDOuRRMxDnP+T7gAAAAQO7G79vkAAAAwDkJcAAAAEhAgAMAAEACAhwAAAASmDQBvnnz5qiuro7S0tKoqamJ3bt3f+j8nTt3Rk1NTZSWlsZ1110XTzzxRKKdQv5yOecvvvhi3HrrrfGpT30qysrKYvHixfGLX/wi4W4hP7n+Pj/rtddei6KiovjiF784sRuEcZDrOR8aGooNGzZEVVVVlJSUxGc+85nYtm1bot1CfnI959u3b4/58+fH5ZdfHpWVlXHPPfdEf39/ot1Cbnbt2hXLly+PWbNmRUFBQbz88ssfuWY8GnRSBHhbW1usWbMmNmzYEF1dXVFfXx9Lly4dcTuz/+vIkSOxbNmyqK+vj66urnjggQdi9erV8cILLyTeOZy/XM/5rl274tZbb4329vbo7OyMm266KZYvXx5dXV2Jdw7nL9dzftbAwEA0NTXFV77ylUQ7hfzlc85XrlwZ//mf/xlbt26N//qv/4pnn302brzxxoS7htzkes737NkTTU1NsWrVqnj99dfjueeei9/85jdx7733Jt45nJ9333035s+fH4899th5zR+3Bs0mgYULF2bNzc0jxm688cZs3bp1Y87/zne+k914440jxr7xjW9kX/rSlyZsj3Chcj3nY/n85z+fbdy4cby3BuMm33Pe2NiYfe9738seeuihbP78+RO4Q7hwuZ7zn//851l5eXnW39+fYnswLnI95//6r/+aXXfddSPGHn300Wz27NkTtkcYLxGRvfTSSx86Z7wa9KJfAT958mR0dnZGQ0PDiPGGhobYu3fvmGv27ds3av5tt90W+/fvj/fff3/C9gr5yuec/7UzZ87EiRMn4sorr5yILcIFy/ecP/XUU/HGG2/EQw89NNFbhAuWzzl/5ZVXora2Nn74wx/GNddcEzfccEPcd9998Ze//CXFliFn+Zzzurq6OHbsWLS3t0eWZfHWW2/F888/H7fffnuKLcOEG68GLRrvjeWqr68vTp8+HRUVFSPGKyoqore3d8w1vb29Y84/depU9PX1RWVl5YTtF/KRzzn/az/60Y/i3XffjZUrV07EFuGC5XPO//CHP8S6deti9+7dUVR00f+RBB8pn3N++PDh2LNnT5SWlsZLL70UfX198c1vfjPefvtt3wNnUsrnnNfV1cX27dujsbEx/ud//idOnToV//AP/xA//vGPU2wZJtx4NehFvwJ+VkFBwYifsywbNfZR88cah8kk13N+1rPPPhvf//73o62tLa666qqJ2h6Mi/M956dPn4477rgjNm7cGDfccEOq7cG4yOX3+ZkzZ6KgoCC2b98eCxcujGXLlsUjjzwSTz/9tKvgTGq5nPODBw/G6tWr48EHH4zOzs549dVX48iRI9Hc3Jxiq5DEeDToRb/cMHPmzCgsLBz1b9OOHz8+6t8wnHX11VePOb+oqChmzJgxYXuFfOVzzs9qa2uLVatWxXPPPRe33HLLRG4TLkiu5/zEiROxf//+6Orqim9/+9sR8UGoZFkWRUVFsWPHjrj55puT7B3OVz6/zysrK+Oaa66J8vLy4bG5c+dGlmVx7NixuP766yd0z5CrfM55a2trLFmyJO6///6IiPjCF74QV1xxRdTX18fDDz/sE6pMeePVoBf9CnhxcXHU1NRER0fHiPGOjo6oq6sbc83ixYtHzd+xY0fU1tbG9OnTJ2yvkK98znnEB1e+77777njmmWd8h4pJL9dzXlZWFr/73e/iwIEDw4/m5ub43Oc+FwcOHIhFixal2jqct3x+ny9ZsiT+9Kc/xTvvvDM89vvf/z6mTZsWs2fPntD9Qj7yOefvvfdeTJs2Mi0KCwsj4v9fJYSpbNwaNKc/2TZB/uM//iObPn16tnXr1uzgwYPZmjVrsiuuuCL77//+7yzLsmzdunXZnXfeOTz/8OHD2eWXX56tXbs2O3jwYLZ169Zs+vTp2fPPP3+x3gJ8pFzP+TPPPJMVFRVljz/+eNbT0zP8+POf/3yx3gJ8pFzP+V/zV9CZCnI95ydOnMhmz56d/eM//mP2+uuvZzt37syuv/767N57771YbwE+Uq7n/KmnnsqKioqyzZs3Z2+88Ua2Z8+erLa2Nlu4cOHFegvwoU6cOJF1dXVlXV1dWURkjzzySNbV1ZW9+eabWZZNXINOigDPsix7/PHHs6qqqqy4uDhbsGBBtnPnzuH/7K677sq+/OUvj5j/q1/9Kvu7v/u7rLi4OPv0pz+dbdmyJfGOIXe5nPMvf/nLWUSMetx1113pNw45yPX3+f8lwJkqcj3nhw4dym655Zbssssuy2bPnp21tLRk7733XuJdQ25yPeePPvpo9vnPfz677LLLssrKyuzrX/96duzYscS7hvPzy1/+8kP/v/ZENWhBlvlMCAAAAEy0i/4dcAAAALgUCHAAAABIQIADAABAAgIcAAAAEhDgAAAAkIAABwAAgAQEOAAAACQgwAEAACABAQ4AAAAJCHAAAABIQIADAABAAgIcAAAAEvhfvPOSx8TMqn4AAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 1200x500 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "fig, ax = plt.subplots(1,1,figsize=(12,5))  #for barplots yields/excretion\n",
    "plot_eteTree.plot_tree(t, align_names=False, name_offset=None, max_dist=None, font_size=9, axe=ax,font_size2=8,alternative_names=True,short_names=False)\n",
    "lc=-1\n",
    "for level in [\"phylum\",\"class\",\"order\",\"family\",\"genus\",\"species\"]:\n",
    "    lc=lc+1\n",
    "    lpos=0.5+lc*branchdistance_number\n",
    "    ax.text(lpos,-1.2,level,fontsize=14)\n",
    "    ax.text(lpos,-2.5,\"(\"+str(round(abundancevalues_sum[lc],1))+\"%)\",fontsize=14,color='k')\n",
    "\n",
    "ax.text(6,-4,\"taxonomic rank\")\n",
    "fig.tight_layout()\n",
    "fig.savefig(\"plot_output_figure1/tree_matplotlib.pdf\")\n",
    "plt.show()\n",
    "\n",
    "###### old, using eTe3 plotting options\n",
    "\n",
    "#decide what to add to nodes\n",
    "def my_layout(node):\n",
    "        F = TextFace(node.name[:5]) #tight_text=True\n",
    "        A = TextFace(node.abundance) #, tight_text=True \n",
    "        #print(node)\n",
    "        if node.is_leaf():\n",
    "            add_face_to_node(F, node, column=1, position=\"branch-top\")\n",
    "            #add_face_to_node(F, node, column=0,position='aligned')\n",
    "            add_face_to_node(A, node, column=0,position='aligned')\n",
    "        else:\n",
    "            add_face_to_node(F, node, column=0, position=\"branch-top\")\n",
    "            add_face_to_node(A, node, column=0, position=\"branch-bottom\")\n",
    "\n",
    "            \n",
    "            #if node.is_leaf():\n",
    "        #        seq_face = SeqMotifFace(node.sequence, seqtype='aa', seq_format='seq')\n",
    "        #        add_face_to_node(seq_face, node, column=0, position='aligned')\n",
    "    \n",
    "ts.layout_fn = my_layout\n",
    "\n",
    "#ts.show_branch_length = True\n",
    "#ts.show_branch_support = True\n",
    "#t.render(\"plot_output_Fig2/tree.pdf\", w=8, units=\"in\", tree_style=ts)\n",
    "\n",
    "#t.show(tree_style=ts)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plot compact tree without full names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'plot_eteTree' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[15], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m fig, ax \u001b[38;5;241m=\u001b[39m plt\u001b[38;5;241m.\u001b[39msubplots(\u001b[38;5;241m1\u001b[39m,\u001b[38;5;241m1\u001b[39m,figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m4\u001b[39m,\u001b[38;5;241m5\u001b[39m))  \u001b[38;5;66;03m#for barplots yields/excretion\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m \u001b[43mplot_eteTree\u001b[49m\u001b[38;5;241m.\u001b[39mplot_tree(t, align_names\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, name_offset\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, max_dist\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, font_size\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m9\u001b[39m, axe\u001b[38;5;241m=\u001b[39max,font_size2\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m8\u001b[39m,alternative_names\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,short_names\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,plot_abundancenumbers\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,abundancestyle\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mplain\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m      3\u001b[0m lc\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m level \u001b[38;5;129;01min\u001b[39;00m [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mphylum\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mclass\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124morder\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfamily\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgenus\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mspecies\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n",
      "\u001b[0;31mNameError\u001b[0m: name 'plot_eteTree' is not defined"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAGyCAYAAAD00j/6AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjYuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/P9b71AAAACXBIWXMAAA9hAAAPYQGoP6dpAAAaXklEQVR4nO3de2xUdf7/8de0pVNgt2MELS3UWlzQKhGXNlTKNkYXa4BgSNxQ48aii4mNuly6uFK7ESEmjW4kK0rrrZWYVLbxguGPrjJ/7EK57IVua4xtohGWFm1pWuK0XrZI+Xz/4NfxN7awnOlMC+8+H8n8MR8/Z+bzsfr0eHpy8DnnnAAAl72E8V4AACA2CDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAY4Tno+/fv14oVK5SRkSGfz6f333//fx6zb98+5ebmKiUlRbNnz9bLL78czVoBABfgOejffPON5s+fr5deeumi5h87dkzLli1TYWGhmpub9eSTT2rt2rV69913PS8WAHB+vtE8nMvn82n37t1auXLleec88cQT2rNnj9ra2sJjpaWl+uijj3T48OFovxoA8CNJ8f6Cw4cPq6ioKGLsrrvuUk1Njb7//ntNmjRp2DEDAwMaGBgIvz979qxOnTqladOmyefzxXvJABB3zjn19/crIyNDCQmx+XVm3IPe1dWltLS0iLG0tDSdOXNGPT09Sk9PH3ZMZWWltmzZEu+lAcC46+jo0KxZs2LyWXEPuqRhZ9VDV3nOd7ZdXl6usrKy8PtQKKRrrrlGHR0dSk1Njd9CAWCM9PX1KTMzUz/96U9j9plxD/qMGTPU1dUVMdbd3a2kpCRNmzZtxGP8fr/8fv+w8dTUVIIOwJRYXkaO+33oixYtUjAYjBjbu3ev8vLyRrx+DgCIjuegf/3112ppaVFLS4ukc7cltrS0qL29XdK5yyUlJSXh+aWlpTp+/LjKysrU1tam2tpa1dTUaOPGjbHZAQBAUhSXXI4cOaLbb789/H7oWvfq1au1c+dOdXZ2huMuSdnZ2WpoaNCGDRu0Y8cOZWRkaPv27brnnntisHwAwJBR3Yc+Vvr6+hQIBBQKhbiGDsCEeHSNZ7kAgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjIgq6FVVVcrOzlZKSopyc3PV2Nh4wfl1dXWaP3++pkyZovT0dD344IPq7e2NasEAgJF5Dnp9fb3Wr1+viooKNTc3q7CwUEuXLlV7e/uI8w8cOKCSkhKtWbNGn3zyid5++23961//0kMPPTTqxQMAfuA56Nu2bdOaNWv00EMPKScnR3/605+UmZmp6urqEef//e9/17XXXqu1a9cqOztbv/jFL/Twww/ryJEjo148AOAHnoJ++vRpNTU1qaioKGK8qKhIhw4dGvGYgoICnThxQg0NDXLO6eTJk3rnnXe0fPny837PwMCA+vr6Il4AgAvzFPSenh4NDg4qLS0tYjwtLU1dXV0jHlNQUKC6ujoVFxcrOTlZM2bM0BVXXKEXX3zxvN9TWVmpQCAQfmVmZnpZJgBMSFH9UtTn80W8d84NGxvS2tqqtWvX6qmnnlJTU5M++OADHTt2TKWlpef9/PLycoVCofCro6MjmmUCwISS5GXy9OnTlZiYOOxsvLu7e9hZ+5DKykotXrxYjz/+uCTp5ptv1tSpU1VYWKhnnnlG6enpw47x+/3y+/1elgYAE56nM/Tk5GTl5uYqGAxGjAeDQRUUFIx4zLfffquEhMivSUxMlHTuzB4AEBueL7mUlZXp9ddfV21trdra2rRhwwa1t7eHL6GUl5erpKQkPH/FihV67733VF1draNHj+rgwYNau3atFi5cqIyMjNjtBAAmOE+XXCSpuLhYvb292rp1qzo7OzVv3jw1NDQoKytLktTZ2RlxT/oDDzyg/v5+vfTSS/rd736nK664QnfccYeeffbZ2O0CACCfuwyue/T19SkQCCgUCik1NXW8lwMAoxaPrvEsFwAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARUQW9qqpK2dnZSklJUW5urhobGy84f2BgQBUVFcrKypLf79d1112n2traqBYMABhZktcD6uvrtX79elVVVWnx4sV65ZVXtHTpUrW2tuqaa64Z8ZhVq1bp5MmTqqmp0c9+9jN1d3frzJkzo148AOAHPuec83JAfn6+FixYoOrq6vBYTk6OVq5cqcrKymHzP/jgA9177706evSorrzyyqgW2dfXp0AgoFAopNTU1Kg+AwAuJfHomqdLLqdPn1ZTU5OKiooixouKinTo0KERj9mzZ4/y8vL03HPPaebMmZo7d642btyo77777rzfMzAwoL6+vogXAODCPF1y6enp0eDgoNLS0iLG09LS1NXVNeIxR48e1YEDB5SSkqLdu3erp6dHjzzyiE6dOnXe6+iVlZXasmWLl6UBwIQX1S9FfT5fxHvn3LCxIWfPnpXP51NdXZ0WLlyoZcuWadu2bdq5c+d5z9LLy8sVCoXCr46OjmiWCQATiqcz9OnTpysxMXHY2Xh3d/ews/Yh6enpmjlzpgKBQHgsJydHzjmdOHFCc+bMGXaM3++X3+/3sjQAmPA8naEnJycrNzdXwWAwYjwYDKqgoGDEYxYvXqwvv/xSX3/9dXjs008/VUJCgmbNmhXFkgEAI/F8yaWsrEyvv/66amtr1dbWpg0bNqi9vV2lpaWSzl0uKSkpCc+/7777NG3aND344INqbW3V/v379fjjj+s3v/mNJk+eHLudAMAE5/k+9OLiYvX29mrr1q3q7OzUvHnz1NDQoKysLElSZ2en2tvbw/N/8pOfKBgM6re//a3y8vI0bdo0rVq1Ss8880zsdgEA8H4f+njgPnQA1oz7fegAgEsXQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACOiCnpVVZWys7OVkpKi3NxcNTY2XtRxBw8eVFJSkm655ZZovhYAcAGeg15fX6/169eroqJCzc3NKiws1NKlS9Xe3n7B40KhkEpKSvTLX/4y6sUCAM7P55xzXg7Iz8/XggULVF1dHR7LycnRypUrVVlZed7j7r33Xs2ZM0eJiYl6//331dLSctHf2dfXp0AgoFAopNTUVC/LBYBLUjy65ukM/fTp02pqalJRUVHEeFFRkQ4dOnTe49544w19/vnn2rx580V9z8DAgPr6+iJeAIAL8xT0np4eDQ4OKi0tLWI8LS1NXV1dIx7z2WefadOmTaqrq1NSUtJFfU9lZaUCgUD4lZmZ6WWZADAhRfVLUZ/PF/HeOTdsTJIGBwd13333acuWLZo7d+5Ff355eblCoVD41dHREc0yAWBCubhT5v9n+vTpSkxMHHY23t3dPeysXZL6+/t15MgRNTc367HHHpMknT17Vs45JSUlae/evbrjjjuGHef3++X3+70sDQAmPE9n6MnJycrNzVUwGIwYDwaDKigoGDY/NTVVH3/8sVpaWsKv0tJSXX/99WppaVF+fv7oVg8ACPN0hi5JZWVluv/++5WXl6dFixbp1VdfVXt7u0pLSyWdu1zyxRdf6M0331RCQoLmzZsXcfzVV1+tlJSUYeMAgNHxHPTi4mL19vZq69at6uzs1Lx589TQ0KCsrCxJUmdn5/+8Jx0AEHue70MfD9yHDsCacb8PHQBw6SLoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgRFRBr6qqUnZ2tlJSUpSbm6vGxsbzzn3vvfd055136qqrrlJqaqoWLVqkDz/8MOoFAwBG5jno9fX1Wr9+vSoqKtTc3KzCwkItXbpU7e3tI87fv3+/7rzzTjU0NKipqUm33367VqxYoebm5lEvHgDwA59zznk5ID8/XwsWLFB1dXV4LCcnRytXrlRlZeVFfcZNN92k4uJiPfXUUxc1v6+vT4FAQKFQSKmpqV6WCwCXpHh0zdMZ+unTp9XU1KSioqKI8aKiIh06dOiiPuPs2bPq7+/XlVdeed45AwMD6uvri3gBAC7MU9B7eno0ODiotLS0iPG0tDR1dXVd1Gc8//zz+uabb7Rq1arzzqmsrFQgEAi/MjMzvSwTACakqH4p6vP5It4754aNjWTXrl16+umnVV9fr6uvvvq888rLyxUKhcKvjo6OaJYJABNKkpfJ06dPV2Ji4rCz8e7u7mFn7T9WX1+vNWvW6O2339aSJUsuONfv98vv93tZGgBMeJ7O0JOTk5Wbm6tgMBgxHgwGVVBQcN7jdu3apQceeEBvvfWWli9fHt1KAQAX5OkMXZLKysp0//33Ky8vT4sWLdKrr76q9vZ2lZaWSjp3ueSLL77Qm2++KelczEtKSvTCCy/o1ltvDZ/dT548WYFAIIZbAYCJzXPQi4uL1dvbq61bt6qzs1Pz5s1TQ0ODsrKyJEmdnZ0R96S/8sorOnPmjB599FE9+uij4fHVq1dr586do98BAEBSFPehjwfuQwdgzbjfhw4AuHQRdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMIKgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYARBBwAjCDoAGEHQAcAIgg4ARhB0ADCCoAOAEQQdAIwg6ABgBEEHACMIOgAYQdABwAiCDgBGEHQAMCKqoFdVVSk7O1spKSnKzc1VY2PjBefv27dPubm5SklJ0ezZs/Xyyy9HtVgAwPl5Dnp9fb3Wr1+viooKNTc3q7CwUEuXLlV7e/uI848dO6Zly5apsLBQzc3NevLJJ7V27Vq9++67o148AOAHPuec83JAfn6+FixYoOrq6vBYTk6OVq5cqcrKymHzn3jiCe3Zs0dtbW3hsdLSUn300Uc6fPjwRX1nX1+fAoGAQqGQUlNTvSwXAC5J8ehakpfJp0+fVlNTkzZt2hQxXlRUpEOHDo14zOHDh1VUVBQxdtddd6mmpkbff/+9Jk2aNOyYgYEBDQwMhN+HQiFJ5/4GAIAFQz3zeE59QZ6C3tPTo8HBQaWlpUWMp6Wlqaura8Rjurq6Rpx/5swZ9fT0KD09fdgxlZWV2rJly7DxzMxML8sFgEteb2+vAoFATD7LU9CH+Hy+iPfOuWFj/2v+SONDysvLVVZWFn7/1VdfKSsrS+3t7THb+OWgr69PmZmZ6ujomFCXmtg3+54IQqGQrrnmGl155ZUx+0xPQZ8+fboSExOHnY13d3cPOwsfMmPGjBHnJyUladq0aSMe4/f75ff7h40HAoEJ9QMfkpqayr4nEPY9sSQkxO7ucU+flJycrNzcXAWDwYjxYDCogoKCEY9ZtGjRsPl79+5VXl7eiNfPAQDR8fyfhrKyMr3++uuqra1VW1ubNmzYoPb2dpWWlko6d7mkpKQkPL+0tFTHjx9XWVmZ2traVFtbq5qaGm3cuDF2uwAAeL+GXlxcrN7eXm3dulWdnZ2aN2+eGhoalJWVJUnq7OyMuCc9OztbDQ0N2rBhg3bs2KGMjAxt375d99xzz0V/p9/v1+bNm0e8DGMZ+2bfEwH7jt2+Pd+HDgC4NPEsFwAwgqADgBEEHQCMIOgAYMQlE/SJ+kheL/t+7733dOedd+qqq65SamqqFi1apA8//HAMVxs7Xn/eQw4ePKikpCTdcsst8V1gnHjd98DAgCoqKpSVlSW/36/rrrtOtbW1Y7Ta2PG677q6Os2fP19TpkxRenq6HnzwQfX29o7Rakdv//79WrFihTIyMuTz+fT+++//z2Ni0jR3Cfjzn//sJk2a5F577TXX2trq1q1b56ZOneqOHz8+4vyjR4+6KVOmuHXr1rnW1lb32muvuUmTJrl33nlnjFc+Ol73vW7dOvfss8+6f/7zn+7TTz915eXlbtKkSe7f//73GK98dLzue8hXX33lZs+e7YqKitz8+fPHZrExFM2+7777bpefn++CwaA7duyY+8c//uEOHjw4hqsePa/7bmxsdAkJCe6FF15wR48edY2Nje6mm25yK1euHOOVR6+hocFVVFS4d99910lyu3fvvuD8WDXtkgj6woULXWlpacTYDTfc4DZt2jTi/N///vfuhhtuiBh7+OGH3a233hq3NcaD132P5MYbb3RbtmyJ9dLiKtp9FxcXuz/84Q9u8+bNl2XQve77L3/5iwsEAq63t3cslhc3Xvf9xz/+0c2ePTtibPv27W7WrFlxW2M8XUzQY9W0cb/kMvRI3h8/YjeaR/IeOXJE33//fdzWGkvR7PvHzp49q/7+/pg+3Cfeot33G2+8oc8//1ybN2+O9xLjIpp979mzR3l5eXruuec0c+ZMzZ07Vxs3btR33303FkuOiWj2XVBQoBMnTqihoUHOOZ08eVLvvPOOli9fPhZLHhexalpUT1uMpbF6JO+lJpp9/9jzzz+vb775RqtWrYrHEuMimn1/9tln2rRpkxobG5WUNO7/yEYlmn0fPXpUBw4cUEpKinbv3q2enh498sgjOnXq1GVzHT2afRcUFKiurk7FxcX673//qzNnzujuu+/Wiy++OBZLHhexatq4n6EPifcjeS9VXvc9ZNeuXXr66adVX1+vq6++Ol7Li5uL3ffg4KDuu+8+bdmyRXPnzh2r5cWNl5/32bNn5fP5VFdXp4ULF2rZsmXatm2bdu7ceVmdpUve9t3a2qq1a9fqqaeeUlNTkz744AMdO3Ys/Lwoq2LRtHE/3RmrR/JeaqLZ95D6+nqtWbNGb7/9tpYsWRLPZcac13339/fryJEjam5u1mOPPSbpXOicc0pKStLevXt1xx13jMnaRyOan3d6erpmzpwZ8WcA5OTkyDmnEydOaM6cOXFdcyxEs+/KykotXrxYjz/+uCTp5ptv1tSpU1VYWKhnnnnmsvg/cK9i1bRxP0OfqI/kjWbf0rkz8wceeEBvvfXWZXlN0eu+U1NT9fHHH6ulpSX8Ki0t1fXXX6+Wlhbl5+eP1dJHJZqf9+LFi/Xll1/q66+/Do99+umnSkhI0KxZs+K63liJZt/ffvvtsGeEJyYmSortH9d2KYlZ0zz9CjVOhm5rqqmpca2trW79+vVu6tSp7j//+Y9zzrlNmza5+++/Pzx/6BafDRs2uNbWVldTU3NZ37Z4sft+6623XFJSktuxY4fr7OwMv7766qvx2kJUvO77xy7Xu1y87ru/v9/NmjXL/epXv3KffPKJ27dvn5szZ4576KGHxmsLUfG67zfeeMMlJSW5qqoq9/nnn7sDBw64vLw8t3DhwvHagmf9/f2uubnZNTc3O0lu27Ztrrm5OXyrZryadkkE3TnnduzY4bKyslxycrJbsGCB27dvX/ivrV692t12220R8//2t7+5n//85y45Odlde+21rrq6eoxXHBte9n3bbbc5ScNeq1evHvuFj5LXn/f/73INunPe993W1uaWLFniJk+e7GbNmuXKysrct99+O8arHj2v+96+fbu78cYb3eTJk116err79a9/7U6cODHGq47eX//61wv+uxqvpvH4XAAwYtyvoQMAYoOgA4ARBB0AjCDoAGAEQQcAIwg6ABhB0AHACIIOAEYQdAAwgqADgBEEHQCMIOgAYMT/Aa7Vee2JrHG3AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 400x500 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig, ax = plt.subplots(1,1,figsize=(4,5))  #for barplots yields/excretion\n",
    "plot_eteTree.plot_tree(t, align_names=False, name_offset=None, max_dist=None, font_size=9, axe=ax,font_size2=8,alternative_names=False,short_names=True,plot_abundancenumbers=True,abundancestyle=\"plain\")\n",
    "lc=-1\n",
    "for level in [\"phylum\",\"class\",\"order\",\"family\",\"genus\",\"species\"]:\n",
    "    lc=lc+1\n",
    "    lpos=0.5+lc*branchdistance_number\n",
    "    \n",
    "    #ax.text(lpos,-1.5,level,fontsize=10)\n",
    "    #ax.text(lpos,-2.5,str(round(abundancevalues_sum[lc],1)),fontsize=10,color='k')\n",
    "\n",
    "ax.text(6,-4,\"taxonomic rank\")\n",
    "fig.tight_layout()\n",
    "fig.savefig(\"plot_output_figure1/tree_matplotlib_compact.pdf\")\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare plots for paper\n",
    "\n",
    "This script takes data from analysis_out.csv which is generated in the final_analysis notebook. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0.2</th>\n",
       "      <th>Unnamed: 0.1</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>notes</th>\n",
       "      <th>strain</th>\n",
       "      <th>species</th>\n",
       "      <th>experiment</th>\n",
       "      <th>experiment_short</th>\n",
       "      <th>exp_number</th>\n",
       "      <th>medium</th>\n",
       "      <th>...</th>\n",
       "      <th>succinate_std</th>\n",
       "      <th>lactate_std</th>\n",
       "      <th>butyrate_std</th>\n",
       "      <th>formate_std</th>\n",
       "      <th>maltose_std</th>\n",
       "      <th>total_ferm</th>\n",
       "      <th>total_ferm_std</th>\n",
       "      <th>total_uptake</th>\n",
       "      <th>total_uptake_std</th>\n",
       "      <th>growth_rate_std</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM20083</td>\n",
       "      <td>B.adolescentis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stan_BHI_30Dec35</td>\n",
       "      <td>199</td>\n",
       "      <td>BHI</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM20083</td>\n",
       "      <td>B.adolescentis</td>\n",
       "      <td>analysis_Zur_Newruns_2023_June</td>\n",
       "      <td>m1</td>\n",
       "      <td>33</td>\n",
       "      <td>YCA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM20083</td>\n",
       "      <td>B.adolescentis</td>\n",
       "      <td>analysis_Zur_Newruns_2023_June</td>\n",
       "      <td>m1</td>\n",
       "      <td>51</td>\n",
       "      <td>YCA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>143.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM20083</td>\n",
       "      <td>B.adolescentis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stan_BHI_30Dec34</td>\n",
       "      <td>198</td>\n",
       "      <td>BHI</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM20083</td>\n",
       "      <td>B.adolescentis</td>\n",
       "      <td>analysis_Zur_Newruns_2023_June</td>\n",
       "      <td>m1</td>\n",
       "      <td>42</td>\n",
       "      <td>YCA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0.2  Unnamed: 0.1  Unnamed: 0  notes    strain         species  \\\n",
       "0             0           0.0       144.0    NaN  DSM20083  B.adolescentis   \n",
       "1             1           1.0        29.0    NaN  DSM20083  B.adolescentis   \n",
       "2             2           2.0        45.0    NaN  DSM20083  B.adolescentis   \n",
       "3             3           3.0       143.0    NaN  DSM20083  B.adolescentis   \n",
       "4             4           4.0        38.0    NaN  DSM20083  B.adolescentis   \n",
       "\n",
       "                       experiment  experiment_short exp_number medium  ...  \\\n",
       "0                             NaN  stan_BHI_30Dec35        199    BHI  ...   \n",
       "1  analysis_Zur_Newruns_2023_June                m1         33    YCA  ...   \n",
       "2  analysis_Zur_Newruns_2023_June                m1         51    YCA  ...   \n",
       "3                             NaN  stan_BHI_30Dec34        198    BHI  ...   \n",
       "4  analysis_Zur_Newruns_2023_June                m1         42    YCA  ...   \n",
       "\n",
       "  succinate_std  lactate_std  butyrate_std formate_std maltose_std total_ferm  \\\n",
       "0           NaN          NaN           NaN         NaN         NaN        NaN   \n",
       "1           NaN          NaN           NaN         NaN         NaN        NaN   \n",
       "2           NaN          NaN           NaN         NaN         NaN        NaN   \n",
       "3           NaN          NaN           NaN         NaN         NaN        NaN   \n",
       "4           NaN          NaN           NaN         NaN         NaN        NaN   \n",
       "\n",
       "   total_ferm_std  total_uptake  total_uptake_std  growth_rate_std  \n",
       "0             NaN           NaN               NaN              NaN  \n",
       "1             NaN           NaN               NaN              NaN  \n",
       "2             NaN           NaN               NaN              NaN  \n",
       "3             NaN           NaN               NaN              NaN  \n",
       "4             NaN           NaN               NaN              NaN  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>species_HPLCname</th>\n",
       "      <th>species</th>\n",
       "      <th>new_species</th>\n",
       "      <th>species.1</th>\n",
       "      <th>species_short</th>\n",
       "      <th>new_genus</th>\n",
       "      <th>genus</th>\n",
       "      <th>new_family</th>\n",
       "      <th>family</th>\n",
       "      <th>new_order</th>\n",
       "      <th>order</th>\n",
       "      <th>new_class</th>\n",
       "      <th>class</th>\n",
       "      <th>new_phylum</th>\n",
       "      <th>phylum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B.vulgatus</td>\n",
       "      <td>Phocaeicola vulgatus</td>\n",
       "      <td>Phocaeicola vulgatus</td>\n",
       "      <td>Bacteroides vulgatus</td>\n",
       "      <td>B. vulgatus</td>\n",
       "      <td>Phocaeicola</td>\n",
       "      <td>Phocaeicola</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B.fragilis</td>\n",
       "      <td>Bacteroides fragilis</td>\n",
       "      <td>Bacteroides fragilis</td>\n",
       "      <td>Bacteroides fragilis</td>\n",
       "      <td>B. fragilis</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>B.ovatus</td>\n",
       "      <td>Bacteroides ovatus</td>\n",
       "      <td>Bacteroides ovatus</td>\n",
       "      <td>Bacteroides ovatus</td>\n",
       "      <td>B. ovatus</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B.theta</td>\n",
       "      <td>Bacteroides thetaiotaomicron</td>\n",
       "      <td>Bacteroides thetaiotaomicron</td>\n",
       "      <td>Bacteroides thetaiotaomicron</td>\n",
       "      <td>B. theta</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroides</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P.copri</td>\n",
       "      <td>Prevotella copri</td>\n",
       "      <td>Prevotella copri</td>\n",
       "      <td>Prevotella copri</td>\n",
       "      <td>P. copri</td>\n",
       "      <td>Prevotella</td>\n",
       "      <td>Prevotella</td>\n",
       "      <td>Prevotellaceae</td>\n",
       "      <td>Prevotellaceae</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  species_HPLCname                       species  \\\n",
       "0       B.vulgatus          Phocaeicola vulgatus   \n",
       "1       B.fragilis          Bacteroides fragilis   \n",
       "2         B.ovatus            Bacteroides ovatus   \n",
       "3          B.theta  Bacteroides thetaiotaomicron   \n",
       "4          P.copri              Prevotella copri   \n",
       "\n",
       "                    new_species                     species.1 species_short  \\\n",
       "0          Phocaeicola vulgatus          Bacteroides vulgatus   B. vulgatus   \n",
       "1          Bacteroides fragilis          Bacteroides fragilis   B. fragilis   \n",
       "2            Bacteroides ovatus            Bacteroides ovatus     B. ovatus   \n",
       "3  Bacteroides thetaiotaomicron  Bacteroides thetaiotaomicron      B. theta   \n",
       "4              Prevotella copri              Prevotella copri      P. copri   \n",
       "\n",
       "     new_genus        genus      new_family          family      new_order  \\\n",
       "0  Phocaeicola  Phocaeicola  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "1  Bacteroides  Bacteroides  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "2  Bacteroides  Bacteroides  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "3  Bacteroides  Bacteroides  Bacteroidaceae  Bacteroidaceae  Bacteroidales   \n",
       "4   Prevotella   Prevotella  Prevotellaceae  Prevotellaceae  Bacteroidales   \n",
       "\n",
       "           order    new_class        class    new_phylum         phylum  \n",
       "0  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "1  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "2  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "3  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  \n",
       "4  Bacteroidales  Bacteroidia  Bacteroidia  Bacteroidota  Bacteroidetes  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0.2</th>\n",
       "      <th>Unnamed: 0.1</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>notes</th>\n",
       "      <th>strain</th>\n",
       "      <th>species</th>\n",
       "      <th>experiment</th>\n",
       "      <th>experiment_short</th>\n",
       "      <th>exp_number</th>\n",
       "      <th>medium</th>\n",
       "      <th>...</th>\n",
       "      <th>succinate_std</th>\n",
       "      <th>lactate_std</th>\n",
       "      <th>butyrate_std</th>\n",
       "      <th>formate_std</th>\n",
       "      <th>maltose_std</th>\n",
       "      <th>total_ferm</th>\n",
       "      <th>total_ferm_std</th>\n",
       "      <th>total_uptake</th>\n",
       "      <th>total_uptake_std</th>\n",
       "      <th>growth_rate_std</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>63</td>\n",
       "      <td>63.0</td>\n",
       "      <td>92.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM 1447</td>\n",
       "      <td>B.vulgatus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>analysis_stan_BHI_24Dec22.csv</td>\n",
       "      <td>118</td>\n",
       "      <td>BHI</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>62</td>\n",
       "      <td>62.0</td>\n",
       "      <td>145.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM 1447</td>\n",
       "      <td>B.vulgatus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stan_BHI_30Dec36</td>\n",
       "      <td>200</td>\n",
       "      <td>BHI</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>64</td>\n",
       "      <td>64.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM 1447</td>\n",
       "      <td>B.vulgatus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>analysis_stan_BHI_2.5mM_Feb23.csv</td>\n",
       "      <td>170</td>\n",
       "      <td>BHI</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>65</td>\n",
       "      <td>65.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM 1447</td>\n",
       "      <td>B.vulgatus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>analysis_stan_BHI_24Dec22.csv</td>\n",
       "      <td>119</td>\n",
       "      <td>BHI</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>66</td>\n",
       "      <td>66.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DSM 1447</td>\n",
       "      <td>B.vulgatus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stan_YCA_230811_2.5mM</td>\n",
       "      <td>183</td>\n",
       "      <td>YCA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCM3722</td>\n",
       "      <td>E.coli</td>\n",
       "      <td>av [146, 112]_used_only: [73.0, 82.0]</td>\n",
       "      <td>av</td>\n",
       "      <td>av_E.coli_NCM3722_gamma</td>\n",
       "      <td>gamma</td>\n",
       "      <td>...</td>\n",
       "      <td>0.172812</td>\n",
       "      <td>0.004613</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.898094</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.978310</td>\n",
       "      <td>0.363957</td>\n",
       "      <td>-10.867657</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.253183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCM3722</td>\n",
       "      <td>E.coli</td>\n",
       "      <td>av [186, 187, 171]_used_only: [75.0, 76.0, 80.0]</td>\n",
       "      <td>av</td>\n",
       "      <td>av_E.coli_NCM3722_BHI</td>\n",
       "      <td>BHI</td>\n",
       "      <td>...</td>\n",
       "      <td>0.851301</td>\n",
       "      <td>0.055860</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.676450</td>\n",
       "      <td>0.062596</td>\n",
       "      <td>19.659161</td>\n",
       "      <td>6.843173</td>\n",
       "      <td>-2.406571</td>\n",
       "      <td>0.166590</td>\n",
       "      <td>0.338551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>172</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCM3722</td>\n",
       "      <td>E.coli</td>\n",
       "      <td>av [22, 2, 1]_used_only: [77.0, 78.0, 81.0]</td>\n",
       "      <td>av</td>\n",
       "      <td>av_E.coli_NCM3722_YCA</td>\n",
       "      <td>YCA</td>\n",
       "      <td>...</td>\n",
       "      <td>0.397216</td>\n",
       "      <td>0.461203</td>\n",
       "      <td>0.096961</td>\n",
       "      <td>1.132453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>21.430628</td>\n",
       "      <td>2.430948</td>\n",
       "      <td>-3.814797</td>\n",
       "      <td>0.058335</td>\n",
       "      <td>0.112149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>79</td>\n",
       "      <td>79.0</td>\n",
       "      <td>105.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCM3722</td>\n",
       "      <td>E.coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>analysis_stan_simple.csv</td>\n",
       "      <td>146</td>\n",
       "      <td>gamma</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCM3722</td>\n",
       "      <td>E.coli</td>\n",
       "      <td>av [150, 162]_used_only: [74.0, 79.0]</td>\n",
       "      <td>av</td>\n",
       "      <td>av_E.coli_NCM3722_epsilon</td>\n",
       "      <td>epsilon</td>\n",
       "      <td>...</td>\n",
       "      <td>0.578481</td>\n",
       "      <td>0.632041</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.371618</td>\n",
       "      <td>0.028349</td>\n",
       "      <td>20.556771</td>\n",
       "      <td>0.795447</td>\n",
       "      <td>-5.493158</td>\n",
       "      <td>0.613745</td>\n",
       "      <td>0.508830</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0.2  Unnamed: 0.1  Unnamed: 0  notes    strain     species  \\\n",
       "63             63          63.0        92.0    NaN  DSM 1447  B.vulgatus   \n",
       "62             62          62.0       145.0    NaN  DSM 1447  B.vulgatus   \n",
       "64             64          64.0       118.0    NaN  DSM 1447  B.vulgatus   \n",
       "65             65          65.0        93.0    NaN  DSM 1447  B.vulgatus   \n",
       "66             66          66.0       128.0    NaN  DSM 1447  B.vulgatus   \n",
       "..            ...           ...         ...    ...       ...         ...   \n",
       "174           174           NaN         NaN    NaN   NCM3722      E.coli   \n",
       "173           173           NaN         NaN    NaN   NCM3722      E.coli   \n",
       "172           172           NaN         NaN    NaN   NCM3722      E.coli   \n",
       "79             79          79.0       105.0    NaN   NCM3722      E.coli   \n",
       "175           175           NaN         NaN    NaN   NCM3722      E.coli   \n",
       "\n",
       "                                           experiment  \\\n",
       "63                                                NaN   \n",
       "62                                                NaN   \n",
       "64                                                NaN   \n",
       "65                                                NaN   \n",
       "66                                                NaN   \n",
       "..                                                ...   \n",
       "174             av [146, 112]_used_only: [73.0, 82.0]   \n",
       "173  av [186, 187, 171]_used_only: [75.0, 76.0, 80.0]   \n",
       "172       av [22, 2, 1]_used_only: [77.0, 78.0, 81.0]   \n",
       "79                                                NaN   \n",
       "175             av [150, 162]_used_only: [74.0, 79.0]   \n",
       "\n",
       "                      experiment_short                 exp_number   medium  \\\n",
       "63       analysis_stan_BHI_24Dec22.csv                        118      BHI   \n",
       "62                    stan_BHI_30Dec36                        200      BHI   \n",
       "64   analysis_stan_BHI_2.5mM_Feb23.csv                        170      BHI   \n",
       "65       analysis_stan_BHI_24Dec22.csv                        119      BHI   \n",
       "66               stan_YCA_230811_2.5mM                        183      YCA   \n",
       "..                                 ...                        ...      ...   \n",
       "174                                 av    av_E.coli_NCM3722_gamma    gamma   \n",
       "173                                 av      av_E.coli_NCM3722_BHI      BHI   \n",
       "172                                 av      av_E.coli_NCM3722_YCA      YCA   \n",
       "79            analysis_stan_simple.csv                        146    gamma   \n",
       "175                                 av  av_E.coli_NCM3722_epsilon  epsilon   \n",
       "\n",
       "     ... succinate_std  lactate_std  butyrate_std formate_std maltose_std  \\\n",
       "63   ...           NaN          NaN           NaN         NaN         NaN   \n",
       "62   ...           NaN          NaN           NaN         NaN         NaN   \n",
       "64   ...           NaN          NaN           NaN         NaN         NaN   \n",
       "65   ...           NaN          NaN           NaN         NaN         NaN   \n",
       "66   ...           NaN          NaN           NaN         NaN         NaN   \n",
       "..   ...           ...          ...           ...         ...         ...   \n",
       "174  ...      0.172812     0.004613      0.000000    0.898094         NaN   \n",
       "173  ...      0.851301     0.055860      0.000000    4.676450    0.062596   \n",
       "172  ...      0.397216     0.461203      0.096961    1.132453    0.000000   \n",
       "79   ...           NaN          NaN           NaN         NaN         NaN   \n",
       "175  ...      0.578481     0.632041      0.000000    0.371618    0.028349   \n",
       "\n",
       "    total_ferm  total_ferm_std  total_uptake  total_uptake_std  \\\n",
       "63         NaN             NaN           NaN               NaN   \n",
       "62         NaN             NaN           NaN               NaN   \n",
       "64         NaN             NaN           NaN               NaN   \n",
       "65         NaN             NaN           NaN               NaN   \n",
       "66         NaN             NaN           NaN               NaN   \n",
       "..         ...             ...           ...               ...   \n",
       "174  34.978310        0.363957    -10.867657               NaN   \n",
       "173  19.659161        6.843173     -2.406571          0.166590   \n",
       "172  21.430628        2.430948     -3.814797          0.058335   \n",
       "79         NaN             NaN           NaN               NaN   \n",
       "175  20.556771        0.795447     -5.493158          0.613745   \n",
       "\n",
       "     growth_rate_std  \n",
       "63               NaN  \n",
       "62               NaN  \n",
       "64               NaN  \n",
       "65               NaN  \n",
       "66               NaN  \n",
       "..               ...  \n",
       "174         0.253183  \n",
       "173         0.338551  \n",
       "172         0.112149  \n",
       "79               NaN  \n",
       "175         0.508830  \n",
       "\n",
       "[189 rows x 64 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "list output\n",
      "[]\n",
      "['av_B.vulgatus_DSM 1447_BHI', 'av_B.fragilis_ATCC25285_BHI', 'av_B.ovatus_ATCC8483_BHI', 'av_B.theta_ATCC29148_BHI', 'av_B.finegoldii_HM-727_BHI', 'av_P.distastonis_HM-169_BHI', 'av_E.rectale_ATCC33656_BHI', 'av_R.intestinalis_DSM14610_BHI', 'av_F.prausnitzii_DSM17677_BHI', 'av_B.longum_DSM20219_BHI', 'av_B.adolescentis_DSM20083_BHI', 'av_C.aerofaciens_DSM3979_BHI', 'av_E.coli_NCM3722_BHI']\n",
      "[]\n",
      "['av_B.vulgatus_DSM 1447_YCA', 'av_B.fragilis_ATCC25285_YCA', 'av_B.ovatus_ATCC8483_YCA', 'av_B.theta_ATCC29148_YCA', 'av_B.finegoldii_HM-727_YCA', 'av_B.uniformis_ATCC8492_YCA', 'av_P.copri_DSM18205_YCA', 'av_P.distastonis_HM-169_YCA', 'av_E.rectale_ATCC33656_YCA', 'av_R.intestinalis_DSM14610_YCA', 'av_F.prausnitzii_DSM17677_YCA', 'av_R.bromii_ATCC27255_YCA', 'av_B.longum_DSM20219_YCA', 'av_B.adolescentis_DSM20083_YCA', 'av_C.aerofaciens_DSM3979_YCA', 'av_E.coli_NCM3722_YCA']\n",
      "['av_B.vulgatus_DSM 1447_YCA', 'av_B.fragilis_ATCC25285_YCA', 'av_B.ovatus_ATCC8483_YCA', 'av_B.theta_ATCC29148_YCA', 'av_B.finegoldii_HM-727_YCA', 'av_B.uniformis_ATCC8492_YCA', 'av_P.copri_DSM18205_YCA', 'av_P.distastonis_HM-169_YCA', 'av_E.rectale_ATCC33656_YCA', 'av_R.intestinalis_DSM14610_YCA', 'av_F.prausnitzii_DSM17677_YCA', 'av_R.bromii_ATCC27255_YCA', 'av_B.longum_DSM20219_YCA', 'av_B.adolescentis_DSM20083_YCA', 'av_C.aerofaciens_DSM3979_YCA', 'av_E.coli_NCM3722_YCA']\n",
      "['glucose', 'maltose', 'acetate', 'butyrate', 'formate', 'lactate', 'propionate', 'succinate']\n"
     ]
    }
   ],
   "source": [
    "###################\n",
    "#decide what to plot\n",
    "####################\n",
    "\n",
    "#use a table with average values  (to generate this table, run Final_analysis_hplcdata.ipynb\n",
    "dataout_av=pd.read_csv(\"data_hplc/analysis_out_av.csv\")\n",
    "display(dataout_av.head())\n",
    "display(speciesinformation.head())\n",
    "\n",
    "#sort list manually \n",
    "sorter=[\"B.vulgatus\",\"B.fragilis\",\"B.ovatus\",\"B.theta\",\"B.finegoldii\",\"B.uniformis\",'P.copri','P.distastonis',\"E.rectale\",\"R.intestinalis\",\"F.prausnitzii\",\"R.bromii\",\"B.longum\",\"B.adolescentis\",\"C.aerofaciens\",\"E.coli\",\"E.coliI\",\"E.coliII\",\"Fecal\",\"FecalPP\",\"ECOR\",\"E.halli\"]\n",
    "dataout_av.sort_values(by=\"species\", key=lambda column: column.map(lambda e: sorter.index(e)), inplace=True)\n",
    "\n",
    "display(dataout_av)\n",
    "\n",
    "#speciesall=speciesinformation[\"species_HPLCname\"]\n",
    "speciesall=[\"B.vulgatus\",\"B.fragilis\",\"B.ovatus\",\"B.theta\",\"B.finegoldii\",\"B.uniformis\",'P.copri','P.distastonis',\"E.rectale\",\"R.intestinalis\",\"F.prausnitzii\",\"R.bromii\",\"B.longum\",\"B.adolescentis\",\"C.aerofaciens\",\"E.coli\"]\n",
    "\n",
    "samplenamelist=[[],[],[],[]]\n",
    "for species in speciesall:\n",
    "    mc=-1\n",
    "    for medium in [\"e\",\"BHI\",\"simple\",\"YCA\"]:\n",
    "            mc=mc+1\n",
    "            selectc=dataout_av.loc[(dataout_av[\"experiment_short\"]==\"av\") & (dataout_av[\"species\"]==species) & (dataout_av[\"medium\"]==medium)]\n",
    "            #print(selectc)\n",
    "            for il in range(0,selectc.shape[0]):\n",
    "                samplenamelist[mc].append(selectc[\"exp_number\"].iloc[il])\n",
    "print(\"list output\")\n",
    "for mc in range(0,len(samplenamelist)):\n",
    "    print(samplenamelist[mc])\n",
    "\n",
    "print(samplenamelist[3])\n",
    "\n",
    "sublistshort=['glu','mal','ace','but','for','lac','pro','suc']\n",
    "\n",
    "\n",
    "#colorlist=['b','b','#1b9e77','#66a61e','#a6761d','#e7298a','#d95f02','#7570b3']\n",
    "\n",
    "\n",
    "#colorlist = met_brewer.met_brew(name=\"Egypt\", n=8, brew_type=\"continuous\")\n",
    "#print(colorlist)\n",
    "\n",
    "colorlist=['#dd5129', '#1e8b99', '#2c7591', '#85635d', '#34a28d', '#fab255', '#acb269', '#5db27d']\n",
    "#['glucose', 'maltose', 'acetate', 'butyrate', 'formate', 'lactate', 'propionate', 'succinate']\n",
    "#sublist=['glucose','maltose','acetate','butyrate','formate','lactate','propionate','succinate'] #skipp ethanol here\n",
    "\n",
    "colorlist=['#dd5129', '#85635d', '#2c7591', '#34a28d', '#fab255','#5db27d', '#1e8b99','#acb269']\n",
    "\n",
    "sublist=['glucose','maltose','acetate','butyrate','formate','lactate','propionate','succinate'] #skipp ethanol here\n",
    "print(sublist)\n",
    "\n",
    "markerlist=['s','h','v','^','<','>','d','o']\n",
    "energycontent=np.array([0.68,1.36,0.21,0.52,0.,.33,0.37,0.36]) #kcal/mmol #the energy per mm for different fermentation products3\n",
    "cfactorlist=[6,12,2,4,1,3,3,4]\n",
    "markerlist=['s','v','^','<','>','d','o','h']\n",
    "   \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plot secrection vs OD - one curve for Fig. 2B."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
